---
title: "Temporal Trial"
author: "Maximilian Fernando Becker"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(cache =TRUE)
```


-----------------
  
This document contains all statistical analyses conducted for the manuscript. Note that due to the random iterative nature of some analyses (such as PERMANOVA, CAP & PCA) some of the figure parameters will change slightly during reanalysis, though core results will remain essentially unchanged. 

All data to reproduce analysis can be found here: `https://github.com/mfbeuq/becker_etal_abovegroundpathogen`

-----------------
  
### Load necessary ecological analysis libraries. 

```{r,cache=TRUE, message=FALSE}
library(phyloseq)
library(microbiome)
library(ggord)
library(metagMisc)
library(ggpubr)
library(FSA)
library(knitr)
library(rmarkdown)
library(ape)
library(vegan)
library(philr)
library(compositions)
library(qiime2R)
library(plyr)
library(dplyr)
library(tidyr)
library(PMCMR)
library(tibble)
library(viridis)
library(gridExtra)
library(AcidPlots)
library(grid)
library(colorRamps)
library(rstatix)
library(dunn.test)
library(pairwiseAdonis)
library(dplyr)
library(ANCOMBC)
library(nlme)
library(tidyverse)
library(compositions)
library(readr)
library(DT)
library(matrixStats)
library(pheatmap)
library(RColorBrewer)
library(dendsort)
library(ComplexHeatmap)
library(circlize)
library(round)
library(lme4)
library(emmeans)
set.seed(225)
```

### Contents of this workspace

```{r,cache=TRUE}
load('temporal_trial.RData')
```

* `ps`: phyloseq object including the metadata, and the ASV table, the taxonomy table and the phylogenetic tree, all three created by QIIME2
* `shannon`: Shannon diversity index data of the entire root microbiota
* `shannon_L`: Shannon diversity index data of the loosely associated (L) root microbiota sub data set
* `shannon_T`: Shannon diversity index data of the tightly associated (T) root microbiota sub data set
* `RA.ord.L`: DEICODE ordination generated by QIIME2 of the loosely associated (L) root microbiota sub data set
* `RA.ord.T`: DEICODE ordination generated by QIIME2 of the tightly associated (T) root microbiota sub data set
* `RA.dist.L`: DEICODE distance matrix generated by QIIME2 of the loosely associated (L) root microbiota sub data set
* `RA.dist.T`: DEICODE distance matrix generated by QIIME2 of the tightly associated (T) root microbiota sub data ses
* `q_meta`: the metadata table 
* `DS`: the disease severity table 
* `deicode_theme`: a theme for generating the plots
* `read_distance`: function for importing a DEICODE distance matrix generated by QIIME2
* `rename.ancombc.output``: a function for renaming taxa in an ANCOM-BC table



--------------------

### Disease severity plot and statistics:

```{r, cache=TRUE, warning = FALSE}
#Format loaded DS table
df2 <- DS %>% 
  tidyr::pivot_longer(
    cols = starts_with("i"), 
    names_to = "Date", 
    values_to = "Leafs",
    values_drop_na = TRUE) %>%
  group_by(Pathogen, Date) %>%
  summarise(mean = mean(Leafs), sd = sd(Leafs), n = n())
df2$Date <- as.numeric(gsub("inf","",as.character(df2$Date)))
df2$Pathogen <- factor(df2$Pathogen, levels=c("K","V","L"))
df2 <- df2[!df2$Pathogen=="K",]

#set breaks 
g.breaks <- c(0,10,20,30,40,50,60,70,80,90,100)

#Plot
figure2 <- ggplot(df2, aes(x=Date, y=mean, group=Pathogen, color=Pathogen)) + 
  geom_errorbar(aes(ymin=mean-sd, ymax=mean+sd), width=.15, size=2, 
                position=position_dodge(0.4)) +
  geom_line(size=2) + geom_point(size=6) + deicode_theme + 
  theme(panel.grid.major =  element_line(colour="grey", size=0.5)) +
  scale_x_continuous(name="Days after inoculation", breaks = g.breaks, limits = c(0,51), expand = c(0, 0)) +
  scale_y_continuous(name="Disease severity", limits = c(0,4.5), expand = c(0,0)) +
  scale_color_manual(values=c("#E69F00","#009E73"), name="Pathogen",
                     labels = c(expression(italic("V. inaequalis")),expression(italic("P. leucotricha")))) +
  theme(legend.text.align = 0, legend.spacing.y = unit(1, 'cm'), legend.key.height = unit(1, 'cm'))
print(figure2)
```


```{r, cache=TRUE, warning = FALSE}
#statistics 
df3 <- DS %>% 
  tidyr::pivot_longer(
    cols = starts_with("i"), 
    names_to = "Date", 
    values_to = "DS",
    values_drop_na = TRUE)

#between timepoints for V. inaequalis
df3 %>%
  filter( Pathogen == "V") %>%
  kruskal_test(Date ~ DS) %>%
  add_column(factor = "VENTIN")
Ventin <- df3 %>%
  filter( Pathogen == "V")
dunn.test(Ventin$DS , Ventin$Date, method="bh", alpha=0.1, list=T)

#between timepoints for P. leucotricha
df3 %>%
  filter( Pathogen == "L") %>%
  kruskal_test(Date ~ DS) %>%
  add_column(factor = "Leuco")
Leuco <- df3 %>%
  filter( Pathogen == "L")
dunn.test(Leuco$DS , Leuco$Date, method="bh", alpha=0.1, list=T)
``` 


### Alpha diversity statistics and plot:

```{r, cache=TRUE, warning = FALSE}
# move the sample names to a new column that matches the metadata and allows them to be merged
shannon.df <- shannon$data %>% rownames_to_column("SampleID") 
tmp <- q_meta %>% 
  left_join(shannon.df) 
# define colors and breaks for plot
col_temp <- c("white","white","red","orange","yellow","purple","blue","lightblue")
g.breaks <- c(0,10,20,30,40,50)
figure3a <- tmp %>%
  filter( Compartment != "B") %>% 
  mutate( compartment = Compartment) %>%
  unite(comp_treat, c("compartment", "Pathogen")) %>%
  filter(!is.na(shannon_entropy)) %>%
  ggplot(aes(x=`Timepoint`, y=shannon_entropy, color=`comp_treat`)) +
  stat_summary(geom="errorbar", fun.data=mean_se, width=1,size=2, position=position_nudge(x = 0, y = 0)) +
  stat_summary(geom="line", fun.data=mean_se, size=2, aes(linetype = Compartment)) +
  stat_summary(geom="point", fun.data=mean_se, size=1.5) +
  scale_x_continuous(name="Days after inoculation", breaks = g.breaks, limits = c(-0.5,50), expand = c(0, 0)) +
  scale_y_continuous(name="Shannon diversity", limits = c(4.1,6.8), expand = c(0,0)) +
  scale_color_manual(name="Treatment", values = c("#000000","#009E73","#E69F00","#000000","#009E73","#E69F00"), 
                     labels = c("Control",expression(italic("P. leucotricha")),expression(italic("V. inaequalis")),
                                "","",""),
                     aesthetics = c("colour")) + deicode_theme +
  theme(panel.grid.major =  element_line(colour="grey", size=0.5)) +
  theme(legend.text.align =0, legend.key.height = unit(0.5, 'cm'))
print(figure3a)
```

The following does the statistical analysis for the Shannon index values of the L-compartment

```{r, cache=TRUE, warning = FALSE}
shannon_L.df <- shannon_L
shannon_L.df$Shannon <- as.numeric(shannon_L.df$shannon_entropy)
mean(shannon_L.df$Shannon) 
sd(shannon_L.df$Shannon) 
shannon_L.df$sd <- as.numeric(substring(shannon_L.df$sd_cat, first=4, last=4))
shannon_L.df$sd

# make GLS model and perform ANOVA
shannon_L.df$days <- as.numeric(gsub("TP_", "", shannon_L.df$TP_cat))

Model <- gls(Shannon ~  Pathogen * days,  na.action = na.omit, data = shannon_L.df, 
          correlation = corCAR1(form =~ 1|days), method = "ML")

summary(Model)
Anova(Model)

# Post Hoc analysis using emmeans
emmeans(Model, specs = pairwise ~ Pathogen)
```

The following does the statistical analysis for the Shannon index values of the T-compartment

```{r, cache=TRUE, warning = FALSE}
shannon_T.df <- shannon_T
shannon_T.df$Shannon <- as.numeric(shannon_T.df$shannon_entropy)
mean(shannon_T.df$Shannon) 
sd(shannon_T.df$Shannon) 
shannon_T.df$sd <- as.numeric(substring(shannon_T.df$sd_cat, first=4, last=4))

# make GLS model and perform ANOVA
shannon_T.df$days <- as.numeric(gsub("TP_", "", shannon_T.df$TP_cat))

Model <- gls(Shannon ~  Pathogen * days,  na.action = na.omit, data = shannon_T.df, 
          correlation = corCAR1(form =~ 1|days), method = "ML")

summary(Model)
Anova(Model)

# Post Hoc analysis using emmeans
emmeans(Model, specs = pairwise ~ Pathogen)
```

--------------------
  
### Beta Diversity:
  
  
##### **Loosely associated root microbiome:**  

First, the distance matrix calculated by QIIME2 needs to be merged with the metadata file:

```{r, cache=TRUE}
RA.dist <- RA.dist.L #using a temporary variable
tmp_order <- colnames(RA.dist)
row.names(q_meta) <- q_meta$SampleID
de <- merge(q_meta, RA.dist, by=0, all=F, all.y=F)
row.names(de) <- de$SampleID
de <- de[ order(match(row.names(de), tmp_order)), ]
RA.dist <- RA.dist[ order(match(row.names(RA.dist), tmp_order)), ]
```

Then, PERMANOVA is calculated on the distance matrix
```{r, cache=TRUE}
adonis2(formula = RA.dist ~ de$Timepoint * de$Pathogen * de$sd_cat, permutations = 999) 
```

Pairwise PERMANOVA is calculated for the treatment factor
```{r, cache=TRUE, error=FALSE, message=FALSE}
pairwise.adonis2(RA.dist ~ Pathogen, data=de)
```

ANOSIM is additionally calculated for each factor individually
```{r, cache=TRUE, error=FALSE, message=FALSE}
anosim(x = RA.dist, grouping =  de$Timepoint, permutations = 999) 
anosim(x = RA.dist, grouping =  de$Pathogen, permutations = 999) 
anosim(x = RA.dist, grouping =  de$sd_cat, permutations = 999) 
anosim(x = RA.dist, grouping =  de$l_max, permutations = 999) 
anosim(x = RA.dist, grouping =  de$h_max, permutations = 999) 
```

The PCoA plot for the L-compartment by using the DEICODE ordination
```{r, cache=TRUE, message = FALSE}
load('temporal_trial.RData')
RA.ord <- RA.ord.L
pl <- RA.ord$data$Vectors %>%
  dplyr::select(SampleID, PC1, PC2) %>%
  left_join(q_meta) %>%
  filter( Compartment == "L") %>%
  filter( Pathogen == "L" | Pathogen == "K" | Pathogen == "V") %>% 
  ggplot(aes(x=PC1, y=PC2, color=Timepoint, shape=Pathogen)) +
  geom_point(alpha=0.9, size=10) + deicode_theme + 
  guides(shape = guide_legend(order = 1, override.aes = list(size=6))) +
  xlab(paste("PC1 - ",format(round(100*RA.ord$data$ProportionExplained[1,1],digits = 1), nsmall = 1),"%")) +
  ylab(paste("PC2 - ",format(round(100*RA.ord$data$ProportionExplained[1,2],digits = 1), nsmall = 1),"%")) +
  geom_hline(yintercept = 0, linetype="dotted") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  scale_shape_manual(values=c(16,17,15), name="Inoculation", labels = c("Control",expression(italic("P. leucotricha")),expression(italic("V. inaequalis")))) +
  scale_color_gradient(name="Timepoint", low = "lightblue", high = "red4",
                       breaks = c(0,12,24,36,48), labels = c("0 Dai","12 Dai","24 Dai","36 Dai","48 Dai"), 
                       limits=c(0,48), guide=guide_colourbar(reverse = TRUE)) +
  theme(legend.text.align = 0, legend.spacing.y = unit(1, 'cm'), legend.key.height = unit(0.5, 'cm'))
```


##### **Tightly associated root microbiome:**  

First, the distance matrix calculated by QIIME2 needs to be merged with the metadata file:

```{r, cache=TRUE}
RA.dist <- RA.dist.T #using a temporary variable
tmp_order <- colnames(RA.dist)
row.names(q_meta) <- q_meta$SampleID
de <- merge(q_meta, RA.dist, by=0, all=F, all.y=F)
row.names(de) <- de$SampleID
de <- de[ order(match(row.names(de), tmp_order)), ]
RA.dist <- RA.dist[ order(match(row.names(RA.dist), tmp_order)), ]
```

Then, PERMANOVA is calculated on the distance matrix
```{r, cache=TRUE}
adonis2(formula = RA.dist ~ de$Timepoint * de$Pathogen * de$sd_cat, permutations = 999) 
```

Pairwise PERMANOVA is calculated for the treatment factor
```{r, cache=TRUE, error=FALSE, message=FALSE}
pairwise.adonis2(RA.dist ~ Pathogen, data=de)
```

ANOSIM is additionally calculated for each factor individually
```{r, cache=TRUE, error=FALSE, message=FALSE}
anosim(x = RA.dist, grouping =  de$Timepoint, permutations = 999) 
anosim(x = RA.dist, grouping =  de$Pathogen, permutations = 999) 
anosim(x = RA.dist, grouping =  de$sd_cat, permutations = 999) 
anosim(x = RA.dist, grouping =  de$l_max, permutations = 999) 
anosim(x = RA.dist, grouping =  de$h_max, permutations = 999) 
```

The PCoA plot for the L-compartment by using the DEICODE ordination
```{r, cache=TRUE, message = FALSE}
load('temporal_trial.RData')
RA.ord <- RA.ord.T
pt <- RA.ord$data$Vectors %>%
  dplyr::select(SampleID, PC1, PC2) %>%
  left_join(q_meta) %>%
  filter( Compartment == "T") %>%
  filter( Pathogen == "L" | Pathogen == "K" | Pathogen == "V") %>% 
  ggplot(aes(x=PC1, y=PC2, color=Timepoint, shape=Pathogen)) +
  geom_point(alpha=0.9, size=10) + deicode_theme + 
  guides(shape = guide_legend(order = 1, override.aes = list(size=6))) +
  xlab(paste("PC1 - ",format(round(100*RA.ord$data$ProportionExplained[1,1],digits = 1), nsmall = 1),"%")) +
  ylab(paste("PC2 - ",format(round(100*RA.ord$data$ProportionExplained[1,2],digits = 1), nsmall = 1),"%")) +
  geom_hline(yintercept = 0, linetype="dotted") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  scale_shape_manual(values=c(16,17,15), name="Inoculation", labels = c("Control",expression(italic("P. leucotricha")),expression(italic("V. inaequalis")))) +
  scale_color_gradient(name="Timepoint", low = "lightblue", high = "red4",
                       breaks = c(0,12,24,36,48), labels = c("0 Dai","12 Dai","24 Dai","36 Dai","48 Dai"), 
                       limits=c(0,48), guide=guide_colourbar(reverse = TRUE)) +
  theme(legend.text.align = 0, legend.spacing.y = unit(1, 'cm'), legend.key.height = unit(0.5, 'cm'))
```

##### **Combine both plots to generate figure 4:**  


```{r, cache=TRUE, warning = FALSE}
legend2 <- cowplot::get_legend(pl) 
c1 <- ggpar(pl, legend = "", legend.title = "", font.x = c(20, "bold"), font.y = c(20, "bold"), font.tickslab = c(18, color="black")) 
c2 <- ggpar(pt, legend = "", legend.title = "", font.x = c(20, "bold"), font.y = c(20, "bold"), font.tickslab = c(18, color="black")) 

figure4 <- ggarrange(c1, NULL, NULL, 
                    legend2, c2, NULL,    
                    widths = c(2.1,1.2), heights = c(1.2, 0.05, 1.2),
                    ncol = 2, nrow = 3)
print(figure4)
```


Pairwise PERMANOVA and dispersion
```{r, cache=TRUE, warning = FALSE}
for (comp in c("L", "T")){
  for (DAI in c("DAI_3","DAI_6","DAI_12","DAI_16","DAI_28","DAI_40","DAI_48")){
    print(paste0("The calculations for the ", comp," ", "compartment at ", DAI))
    cat("\n")
    RA.dist <- read.table(paste0("DAI/",comp,"-",DAI,"/distance-matrix.tsv"))
    tmp_order <- colnames(RA.dist)
    row.names(q_meta) <- q_meta$SampleID
    de <- merge(q_meta, RA.dist, by=0, all=F, all.y=F)
    row.names(de) <- de$SampleID
    de <- de[ order(match(row.names(de), tmp_order)), ]
    RA.dist <- RA.dist[ order(match(row.names(RA.dist), tmp_order)), ]
    
    print(adonis2(formula = RA.dist ~ de$Pathogen * de$sd_cat, permutations = 999))
    p.list <- pairwise.adonis2(RA.dist ~ Pathogen, data=de, nperm = 999)
    
    mod <- betadisper(as.dist(RA.dist), de$Pathogen, sqrt.dist = TRUE)
    print(anova(mod))
    print(TukeyHSD(mod))
    cat("\n")

  }
}
```
##### **Figure 5 A B C:**  

```{r, cache=TRUE, warning = FALSE, error = FALSE}
col.plasma <- plasma(8) #set colors for graph

for (comp in c("L", "T")){
  
  #Non-inoculated control
  path="K"
  RA.dist <- read.table(paste0("figure5_data/",comp,"-",path,"/distance-matrix.tsv"))
  tmp_order <- colnames(RA.dist)
  row.names(q_meta) <- q_meta$SampleID
  de <- merge(q_meta, RA.dist, by=0, all=F, all.y=F)
  row.names(de) <- de$SampleID
  de <- de[ order(match(row.names(de), tmp_order)), ]
  RA.dist <- RA.dist[ order(match(row.names(RA.dist), tmp_order)), ]
  
  adonis2(formula = RA.dist ~ de$TP_cat + de$l_max + de$h_max, permutations = 999)
  PA_list <- pairwise.adonis2(RA.dist ~ TP_cat, data=de)
  dispers <- betadisper(as.dist(RA.dist), de$TP_cat, bias.adjust = TRUE,type = "centroid",  add=TRUE, sqrt.dist = TRUE)
  anova(dispers)
  TukeyHSD(dispers)
  
  result <- c()
  for (i in 2:length(PA_list)) {
    list <- names(print(PA_list))
    comparison <- list[[i]]
    names(PA_list)[i] <- "tmp.df"
    R2 <- PA_list$tmp.df$R2[1]
    p <- PA_list$tmp.df$`Pr(>F)`[1]
    names(PA_list)[i] <- "done"
    tmp <- rbind(comparison,R2,p)
    result <- rbind(result,tmp)
  } 
  print(result)
  rm(result)
  
  RA.ord <- read_qza(paste0("figure5_data/",comp,"-",path,"/RA_ordination.qza"))
  p1 <- RA.ord$data$Vectors %>%
    dplyr::select(SampleID, PC1, PC2) %>%
    left_join(q_meta) %>%
    filter( Compartment == comp) %>%
    filter( Pathogen == path) %>% 
    mutate(TP_cat = fct_relevel(TP_cat, "TP_0", "TP_3", "TP_6", "TP_12", "TP_16", "TP_28", "TP_40", "TP_48")) %>%
    ggplot(aes(x=PC1, y=PC2, color=TP_cat, size=sd_max)) +
    geom_point(alpha=0.9, size=5) + deicode_theme + 
    guides(color = guide_legend(order = 1, override.aes = list(size=12))) +
    geom_hline(yintercept = 0, linetype="dotted") + 
    geom_vline(xintercept = 0, linetype="dotted") +
    xlab(paste("PC1 - ",format(round(100*RA.ord$data$ProportionExplained[1,1],digits = 1), nsmall = 1),"%")) +
    ylab(paste("PC2 - ",format(round(100*RA.ord$data$ProportionExplained[1,2],digits = 1), nsmall = 1),"%")) +
    scale_size(name="Disease Severity") +
    theme(legend.text.align = 0, legend.spacing.y = unit(1, 'cm'), legend.key.height = unit(1, 'cm')) +
    scale_color_manual(name="DAI", values = col.plasma, labels = c("0","3","6","12","16","28","40","48"))
  print(p1)
  
  dbRDA <- capscale(formula = RA.dist ~ de$TP_cat + de$l_max + de$h_max, data = de, comm = RA.dist, na.action = na.omit)
  x <- as.data.frame(scores(dbRDA, display = "sites")) 
  tmp <- merge(q_meta, x, by=0) 
  tmp$TP_cat <- factor(tmp$TP_cat, levels = c("TP_0", "TP_3", "TP_6", "TP_12", "TP_16", "TP_28", "TP_40", "TP_48"))
  row.names(tmp) <- tmp$SampleID
  p2 <- ggplot(tmp, aes(x=CAP1, y=CAP2, color=TP_cat, size=sd_max)) +
    geom_point(alpha=0.9, size=5) + deicode_theme + 
    guides(color = guide_legend(order = 1, override.aes = list(size=12))) +
    geom_hline(yintercept = 0, linetype="dotted") + 
    geom_vline(xintercept = 0, linetype="dotted") +
    xlab(paste("CAP1 - ",format(round(100*summary(dbRDA)$cont$importance[2,1],digits = 1), nsmall = 1),"%")) +
    ylab(paste("CAP2 - ",format(round(100*summary(dbRDA)$cont$importance[2,2],digits = 1), nsmall = 1),"%")) +
    scale_size(name="Disease Severity") +
    theme(legend.text.align = 0, legend.spacing.y = unit(1, 'cm'), legend.key.height = unit(1, 'cm')) +
    scale_color_manual(name="DAI", values = col.plasma, labels = c("0","3","6","12","16","28","40","48"))
  print(p2)
  
  #P. leucotricha
  path="L"
  RA.dist <- read.table(paste0("figure5_data/",comp,"-",path,"/distance-matrix.tsv"))
  tmp_order <- colnames(RA.dist)
  row.names(q_meta) <- q_meta$SampleID
  de <- merge(q_meta, RA.dist, by=0, all=F, all.y=F)
  row.names(de) <- de$SampleID
  de <- de[ order(match(row.names(de), tmp_order)), ]
  RA.dist <- RA.dist[ order(match(row.names(RA.dist), tmp_order)), ]
  
  adonis2(formula = RA.dist ~ de$TP_cat * de$sd_max + de$l_max + de$h_max, permutations = 999)
  PA_list <- pairwise.adonis2(RA.dist ~ TP_cat, data=de)
  dispers <- betadisper(as.dist(RA.dist), de$TP_cat, bias.adjust = TRUE, type = "centroid",  add=TRUE, sqrt.dist = TRUE)
  anova(dispers)
  TukeyHSD(dispers)
  
  result <- c()
  for (i in 2:length(PA_list)) {
    list <- names(print(PA_list))
    comparison <- list[[i]]
    names(PA_list)[i] <- "tmp.df"
    R2 <- PA_list$tmp.df$R2[1]
    p <- PA_list$tmp.df$`Pr(>F)`[1]
    names(PA_list)[i] <- "done"
    tmp <- rbind(comparison,R2,p)
    result <- rbind(result,tmp)
  } 
  print(result)
  rm(result)
  
  RA.ord <- read_qza(paste0("figure5_data/",comp,"-",path,"/RA_ordination.qza"))
  p3 <- RA.ord$data$Vectors %>%
    dplyr::select(SampleID, PC1, PC2) %>%
    left_join(q_meta) %>%
    filter( Compartment == comp) %>%
    filter( Leuco == "YES") %>% 
    mutate(TP_cat = fct_relevel(TP_cat, "TP_0", "TP_3", "TP_6", "TP_12", "TP_16", "TP_28", "TP_40", "TP_48")) %>%
    ggplot(aes(x=PC1, y=PC2, color=TP_cat, size=sd_max)) +
    geom_point(alpha=0.9) + deicode_theme + 
    guides(color = guide_legend(order = 1, override.aes = list(size=12))) +
    geom_hline(yintercept = 0, linetype="dotted") + 
    geom_vline(xintercept = 0, linetype="dotted") +
    xlab(paste("PC1 - ",format(round(100*RA.ord$data$ProportionExplained[1,1],digits = 1), nsmall = 1),"%")) +
    ylab(paste("PC2 - ",format(round(100*RA.ord$data$ProportionExplained[1,2],digits = 1), nsmall = 1),"%")) +
    scale_size(name="Disease Severity",range = c(5,16)) +
    theme(legend.text.align = 0, legend.spacing.y = unit(1, 'cm'), legend.key.height = unit(1, 'cm')) +
    scale_color_manual(name="DAI", values = col.plasma[1:7], labels = c("0","3","6","12","16","28","40"))
  print(p3)
  
  dbRDA <- capscale(formula = RA.dist ~ de$TP_cat + de$sd_max + de$l_max + de$h_max, data = de, comm = RA.dist, na.action = na.omit)
  x <- as.data.frame(scores(dbRDA, display = "sites")) 
  tmp <- merge(q_meta, x, by=0) 
  tmp$TP_cat <- factor(tmp$TP_cat, levels = c("TP_0", "TP_3", "TP_6", "TP_12", "TP_16", "TP_28", "TP_40", "TP_48"))
  row.names(tmp) <- tmp$SampleID
  p4 <- ggplot(tmp, aes(x=CAP1, y=CAP2, color=TP_cat, size=sd_max)) +
    geom_point(alpha=0.9) + deicode_theme + 
    guides(color = guide_legend(order = 1, override.aes = list(size=12))) +
    geom_hline(yintercept = 0, linetype="dotted") + 
    geom_vline(xintercept = 0, linetype="dotted") +
    xlab(paste("CAP1 - ",format(round(100*summary(dbRDA)$cont$importance[2,1],digits = 1), nsmall = 1),"%")) +
    ylab(paste("CAP2 - ",format(round(100*summary(dbRDA)$cont$importance[2,2],digits = 1), nsmall = 1),"%")) +
    scale_size(name="Disease Severity",range = c(5,16)) +
    theme(legend.text.align = 0, legend.spacing.y = unit(1, 'cm'), legend.key.height = unit(1, 'cm')) +
    scale_color_manual(name="DAI", values = col.plasma[1:7], labels = c("0","3","6","12","16","28","40"))
  print(p4)
  
  #V. inaequalis
  path="V"
  RA.dist <- read.table(paste0("figure5_data/",comp,"-",path,"/distance-matrix.tsv"))
  tmp_order <- colnames(RA.dist)
  row.names(q_meta) <- q_meta$SampleID
  de <- merge(q_meta, RA.dist, by=0, all=F, all.y=F)
  row.names(de) <- de$SampleID
  de <- de[ order(match(row.names(de), tmp_order)), ]
  RA.dist <- RA.dist[ order(match(row.names(RA.dist), tmp_order)), ]
  
  adonis2(formula = RA.dist ~ de$TP_cat * de$sd_max + de$l_max + de$h_max, permutations = 999)
  PA_list <- pairwise.adonis2(RA.dist ~ TP_cat, data=de)
  dispers <- betadisper(as.dist(RA.dist), de$TP_cat, bias.adjust = TRUE, type = "centroid",  add=TRUE, sqrt.dist = TRUE)
  anova(dispers)
  TukeyHSD(dispers)
  
  result <- c()
  for (i in 2:length(PA_list)) {
    list <- names(print(PA_list))
    comparison <- list[[i]]
    names(PA_list)[i] <- "tmp.df"
    R2 <- PA_list$tmp.df$R2[1]
    p <- PA_list$tmp.df$`Pr(>F)`[1]
    names(PA_list)[i] <- "done"
    tmp <- rbind(comparison,R2,p)
    result <- rbind(result,tmp)
  } 
  print(result)
  rm(result)
  
  RA.ord <- read_qza(paste0("figure5_data/",comp,"-",path,"/RA_ordination.qza"))
  p5 <- RA.ord$data$Vectors %>%
    dplyr::select(SampleID, PC1, PC2) %>%
    left_join(q_meta) %>%
    filter( Compartment == comp) %>%
    filter( Ventin == "YES") %>% 
    mutate(TP_cat = fct_relevel(TP_cat, "TP_0", "TP_3", "TP_6", "TP_12", "TP_16", "TP_28", "TP_40", "TP_48")) %>%
    ggplot(aes(x=PC1, y=PC2, color=TP_cat, size=sd_max)) +
    geom_point(alpha=0.9) + deicode_theme + 
    guides(color = guide_legend(order = 1, override.aes = list(size=12))) +
    geom_hline(yintercept = 0, linetype="dotted") + 
    geom_vline(xintercept = 0, linetype="dotted") +
    xlab(paste("PC1 - ",format(round(100*RA.ord$data$ProportionExplained[1,1],digits = 1), nsmall = 1),"%")) +
    ylab(paste("PC2 - ",format(round(100*RA.ord$data$ProportionExplained[1,2],digits = 1), nsmall = 1),"%")) +
    scale_size(name="Disease \nSeverity",range = c(5,16)) +
    theme(legend.text.align = 0, legend.spacing.y = unit(1, 'cm'), legend.key.height = unit(1, 'cm')) +
    scale_color_manual(name="DAI", values = col.plasma[1:8], labels = c("0","3","6","12","16","28","40","48")) +
    theme(legend.position = "bottom")
  print(p5)
  
  dbRDA <- capscale(formula = RA.dist ~ de$TP_cat + de$sd_max + de$l_max + de$h_max, data = de, comm = RA.dist, na.action = na.omit)
  x <- as.data.frame(scores(dbRDA, display = "sites")) 
  tmp <- merge(q_meta, x, by=0) 
  tmp$TP_cat <- factor(tmp$TP_cat, levels = c("TP_0", "TP_3", "TP_6", "TP_12", "TP_16", "TP_28", "TP_40", "TP_48"))
  row.names(tmp) <- tmp$SampleID
  p6 <- ggplot(tmp, aes(x=CAP1, y=CAP2, color=TP_cat, size=sd_max)) +
    geom_point(alpha=0.9) + deicode_theme + 
    guides(color = guide_legend(order = 1, override.aes = list(size=12))) +
    geom_hline(yintercept = 0, linetype="dotted") + 
    geom_vline(xintercept = 0, linetype="dotted") +
    xlab(paste("CAP1 - ",format(round(100*summary(dbRDA)$cont$importance[2,1],digits = 1), nsmall = 1),"%")) +
    ylab(paste("CAP2 - ",format(round(100*summary(dbRDA)$cont$importance[2,2],digits = 1), nsmall = 1),"%")) +
    scale_size(name="Disease Severity",range = c(5,16)) +
    theme(legend.text.align = 0, legend.spacing.y = unit(1, 'cm'), legend.key.height = unit(1, 'cm')) +
    scale_color_manual(name="DAI", values = col.plasma[1:8], labels = c("0","3","6","12","16","28","40","48"))
  print(p6)
  
  if(comp=="L"){
    p1_L <- p1
    p3_L <- p3
    p5_L <- p5
  } else {
    print(comp)
  }
  
 }


### PcoA combination plot ---------------------------------------------------------
legend2 <- cowplot::get_legend(p5) 
c1 <- ggpar(p1, legend = "", legend.title = "", font.x = c(20, "bold"), font.y = c(20, "bold"), font.tickslab = c(18, color="black")) 
c2 <- ggpar(p3, legend = "", legend.title = "", font.x = c(20, "bold"), font.y = c(20, "bold"), font.tickslab = c(18, color="black")) 
c3 <- ggpar(p5, legend = "", legend.title = "", font.x = c(20, "bold"), font.y = c(20, "bold"), font.tickslab = c(18, color="black")) 
c4 <- ggpar(p1_L, legend = "", legend.title = "", font.x = c(20, "bold"), font.y = c(20, "bold"), font.tickslab = c(18, color="black")) 
c5 <- ggpar(p3_L, legend = "", legend.title = "", font.x = c(20, "bold"), font.y = c(20, "bold"), font.tickslab = c(18, color="black")) 
c6 <- ggpar(p5_L, legend = "", legend.title = "", font.x = c(20, "bold"), font.y = c(20, "bold"), font.tickslab = c(18, color="black")) 

figure5ABC <- ggarrange(c4, c1, NULL,
                    NULL,NULL,NULL,
                    c5, c2, NULL,
                    NULL,NULL,NULL,
                    c6,c3,
                    widths = c(2,2,1), heights = c(1, 0.1, 1, 0.1, 1),
                    ncol = 3, nrow = 5)

print(figure5ABC)
```



##### **Figure 5 D E F:**  

```{r, cache=TRUE, warning = FALSE, message = FALSE, error = FALSE}
# pairwise adonis heatmap ---------------------------------------------------------

for (comp in c("L", "T")){
  
  #Non-inoculated control
  path="K"
  RA.dist <- read.table(paste0("figure5_data/",comp,"-",path,"/distance-matrix.tsv"))
  tmp_order <- colnames(RA.dist)
  row.names(q_meta) <- q_meta$SampleID
  de <- merge(q_meta, RA.dist, by=0, all=F, all.y=F)
  row.names(de) <- de$SampleID
  de <- de[ order(match(row.names(de), tmp_order)), ]
  RA.dist <- RA.dist[ order(match(row.names(RA.dist), tmp_order)), ]
  
  pairwise.adonis(x=RA.dist, factors=de[,"TP_cat"], perm = 999, p.adjust.m='fdr')
  
  #P. leucotricha
  path="L"
  RA.dist <- read.table(paste0("figure5_data/",comp,"-",path,"/distance-matrix.tsv"))
  tmp_order <- colnames(RA.dist)
  row.names(q_meta) <- q_meta$SampleID
  de <- merge(q_meta, RA.dist, by=0, all=F, all.y=F)
  row.names(de) <- de$SampleID
  de <- de[ order(match(row.names(de), tmp_order)), ]
  RA.dist <- RA.dist[ order(match(row.names(RA.dist), tmp_order)), ]
  
  pairwise.adonis(x=RA.dist, factors=de[,"TP_cat"], perm = 999, p.adjust.m='fdr')
  
  #V. inaequalis
  path="V"
  RA.dist <- read.table(paste0("figure5_data/",comp,"-",path,"/distance-matrix.tsv"))
  tmp_order <- colnames(RA.dist)
  row.names(q_meta) <- q_meta$SampleID
  de <- merge(q_meta, RA.dist, by=0, all=F, all.y=F)
  row.names(de) <- de$SampleID
  de <- de[ order(match(row.names(de), tmp_order)), ]
  RA.dist <- RA.dist[ order(match(row.names(RA.dist), tmp_order)), ]
  
  pairwise.adonis(x=RA.dist, factors=de[,"TP_cat"], perm = 999, p.adjust.m='fdr')
}

### The results were converted in a matrix in an external programm and reloaded into R 
### Those files are not available but can be made from the results created above

### Non-inoculated control ---------------------------------------------------------
sig <- read.delim("figure5_data/adonis_Control_pairwise_date_padj.tsv", header = T)
rownames(sig) <- sig$Timepoint; sig$Timepoint = NULL 
sig <- as.matrix(sig)
colnames(sig) <- tp_list
rownames(sig) <- tp_list
R2 <- read.delim("figure5_data/adonis_Control_pairwise_date_R2.tsv", header = T)
rownames(R2) <- R2$Timepoint; R2$Timepoint = NULL 
R1 <- as.matrix(R2)
colnames(R2) <- tp_list
rownames(R2) <- tp_list

# Add significance level
sig.star <- sig
sig.star[is.na(sig.star)] <- ""
sig.star[sig.star > 0.1] <- ""
sig.star[sig.star > 0.05 & sig.star <= 0.1] <- ""
sig.star[sig.star > 0.01 & sig.star <= 0.05] <- "."
sig.star[sig.star > 0.001 & sig.star <= 0.01] <- "*"

colors <- colorRamp2(c(0,0.33, 0.66,1), c("beige","red3", "darkorchid4", "black"), space = "LAB")

lgd = Legend(at = c(0, 1),  col_fun = colors, 
             title = expression(paste("R"^"2", " value")), title_gp = gpar(fontsize = 26, fontface = "bold"), title_gap = unit(0.5, "cm"),
             labels_gp = gpar(col = "black", fontsize = 26, fontface = "bold"), title_position = "topcenter", 
             grid_height = unit(3, "cm"), legend_width = unit(8, "cm"), direction = "horizontal") 

ht = Heatmap(R2, name = "logfold change",
             column_gap = unit(5, "mm"),
             row_gap = unit(5, "mm"), 
             row_names_gp = gpar(fontsize = 32, fontface = "bold"),
             na_col = "grey30",
             col = colors,
             rect_gp = gpar(col = "white", lwd = 2), #white spaces in between
             border = TRUE, 
             cluster_rows = FALSE, #remove cluster
             cluster_columns = FALSE,
             show_column_dend = FALSE, #remove cluster 
             show_heatmap_legend = FALSE, #remove legend
             row_names_side = "right",
             row_names_max_width = unit(2, "cm"),
             row_names_rot = 0,
             row_names_centered = TRUE,
             row_title_gp = gpar(fontsize = 28),
             row_title_rot = 0,
             column_title = expression(paste("PERMANOVA R"^"2", " value")),
             column_title_gp = gpar(fill = "black", col = "white", fontsize = 28, fontface = "bold", border = "black"), 
             column_title_side = "top",
             column_names_max_height = unit(6, "cm"),
             column_names_gp = gpar(fontsize = 30, fontface = "bold", col = "black"),
             column_names_rot = 90,
             column_names_centered = TRUE,
             show_parent_dend_line = FALSE,
             cell_fun = function(j, i, x, y, width, height, fill) {
               if(sig.star[i, j] == "**")
                 grid.text("**", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
               if(sig.star[i, j] == "*")
                 grid.text("*", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
               if(sig.star[i, j] == ".")
                 grid.text(".", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
               if(sig.star[i, j] == "0")
                 grid.text("", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
             }
)

# Create heatmap plot
draw(ht, padding = unit(c(4, 1, 1, 8), "cm")) # add space for titles
draw(lgd, x = unit(30, "cm"), y = unit(4, "cm"))


### Leucotricha ---------------------------------------------------------
sig <- read.delim("figure5_data/adonis_Leuco_pairwise_date_padj.tsv", header = T)
rownames(sig) <- sig$Timepoint; sig$Timepoint = NULL 
sig <- as.matrix(sig)
colnames(sig) <- tp_list
rownames(sig) <- tp_list
R2 <- read.delim("figure5_data/adonis_Leuco_pairwise_date_R2.tsv", header = T)
rownames(R2) <- R2$Timepoint; R2$Timepoint = NULL 
R2 <- as.matrix(R2)
colnames(R2) <- tp_list
rownames(R2) <- tp_list

sig.star <- sig
sig.star[is.na(sig.star)] <- ""
sig.star[sig.star > 0.1] <- ""
sig.star[sig.star > 0.05 & sig.star <= 0.1] <- ""
sig.star[sig.star > 0.01 & sig.star <= 0.05] <- "."
sig.star[sig.star > 0.001 & sig.star <= 0.01] <- "*"

colors <- colorRamp2(c(0,0.33, 0.66,1), c("beige","red3", "darkorchid4", "black"), space = "LAB")

lgd = Legend(at = c(0, 1),  col_fun = colors, 
             title = expression(paste("R"^"2", " value")), title_gp = gpar(fontsize = 26, fontface = "bold"), title_gap = unit(0.5, "cm"),
             labels_gp = gpar(col = "black", fontsize = 26, fontface = "bold"), title_position = "topcenter", 
             grid_height = unit(3, "cm"), legend_width = unit(8, "cm"), direction = "horizontal") 

ht = Heatmap(R2, name = "logfold change",
             column_gap = unit(5, "mm"),
             row_gap = unit(5, "mm"), 
             row_names_gp = gpar(fontsize = 32, fontface = "bold"),
             na_col = "grey30",
             col = colors,
             rect_gp = gpar(col = "white", lwd = 2), #white spaces in between
             border = TRUE, 
             cluster_rows = FALSE, #remove cluster
             cluster_columns = FALSE,
             show_column_dend = FALSE, #remove cluster 
             show_heatmap_legend = FALSE, #remove legend
             row_names_side = "right",
             row_names_max_width = unit(2, "cm"),
             row_names_rot = 0,
             row_names_centered = TRUE,
             row_title_gp = gpar(fontsize = 28),
             row_title_rot = 0,
             column_title = expression(paste("PERMANOVA R"^"2", " value")),
             column_title_gp = gpar(fill = "black", col = "white", fontsize = 28, fontface = "bold", border = "black"), 
             column_title_side = "top",
             column_names_max_height = unit(6, "cm"),
             column_names_gp = gpar(fontsize = 30, fontface = "bold", col = "black"),
             column_names_rot = 90,
             column_names_centered = TRUE,
             show_parent_dend_line = FALSE,
             cell_fun = function(j, i, x, y, width, height, fill) {
               if(sig.star[i, j] == "**")
                 grid.text("**", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
               if(sig.star[i, j] == "*")
                 grid.text("*", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
               if(sig.star[i, j] == ".")
                 grid.text(".", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
               if(sig.star[i, j] == "0")
                 grid.text("", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
             }
)

# Create heatmap plot
draw(ht, padding = unit(c(4, 1, 1, 8), "cm")) # add space for titles
draw(lgd, x = unit(30, "cm"), y = unit(4, "cm"))


### Venturia ---------------------------------------------------------
sig <- read.delim("figure5_data/adonis_Ventin_pairwise_date_padj.tsv", header = T)
rownames(sig) <- sig$Timepoint; sig$Timepoint = NULL 
sig <- as.matrix(sig)
colnames(sig) <- tp_list
rownames(sig) <- tp_list
R2 <- read.delim("figure5_data/adonis_Ventin_pairwise_date_R2.tsv", header = T)
rownames(R2) <- R2$Timepoint; R2$Timepoint = NULL 
R2 <- as.matrix(R2)
colnames(R2) <- tp_list
rownames(R2) <- tp_list

sig.star <- sig
sig.star[is.na(sig.star)] <- ""
sig.star[sig.star > 0.1] <- ""
sig.star[sig.star > 0.05 & sig.star <= 0.1] <- ""
sig.star[sig.star > 0.01 & sig.star <= 0.05] <- "."
sig.star[sig.star > 0.001 & sig.star <= 0.01] <- "*"

colors <- colorRamp2(c(0,0.33, 0.66,1), c("beige","red3", "darkorchid4", "black"), space = "LAB")

lgd = Legend(at = c(0, 1),  col_fun = colors, 
             title = expression(paste("R"^"2", " value")), title_gp = gpar(fontsize = 26, fontface = "bold"), title_gap = unit(0.5, "cm"),
             labels_gp = gpar(col = "black", fontsize = 26, fontface = "bold"), title_position = "topcenter", 
             grid_height = unit(3, "cm"), legend_width = unit(8, "cm"), direction = "horizontal") 

ht = Heatmap(R2, name = "logfold change",
             column_gap = unit(5, "mm"),
             row_gap = unit(5, "mm"), 
             row_names_gp = gpar(fontsize = 32, fontface = "bold"),
             na_col = "grey30",
             col = colors,
             rect_gp = gpar(col = "white", lwd = 2), #white spaces in between
             border = TRUE, 
             cluster_rows = FALSE, #remove cluster
             cluster_columns = FALSE,
             show_column_dend = FALSE, #remove cluster 
             show_heatmap_legend = FALSE, #remove legend
             row_names_side = "right",
             row_names_max_width = unit(2, "cm"),
             row_names_rot = 0,
             row_names_centered = TRUE,
             row_title_gp = gpar(fontsize = 28),
             row_title_rot = 0,
             column_title = expression(paste("PERMANOVA R"^"2", " value")),
             column_title_gp = gpar(fill = "black", col = "white", fontsize = 28, fontface = "bold", border = "black"), 
             column_title_side = "top",
             column_names_max_height = unit(6, "cm"),
             column_names_gp = gpar(fontsize = 30, fontface = "bold", col = "black"),
             column_names_rot = 90,
             column_names_centered = TRUE,
             show_parent_dend_line = FALSE,
             cell_fun = function(j, i, x, y, width, height, fill) {
               if(sig.star[i, j] == "**")
                 grid.text("**", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
               if(sig.star[i, j] == "*")
                 grid.text("*", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
               if(sig.star[i, j] == ".")
                 grid.text(".", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
               if(sig.star[i, j] == "0")
                 grid.text("", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
             }
)

# Create heatmap plot
draw(ht, padding = unit(c(4, 1, 1, 8), "cm")) # add space for titles
draw(lgd, x = unit(30, "cm"), y = unit(4, "cm"))


```

--------------------
  
### Stacked barplot:
  
##### **Figure S1:**  


```{r, cache=FALSE, warning = FALSE, message = FALSE, error = FALSE}

# Load data, packages and create themes and functions:
load('temporal_trial.RData')

library(plyr)
library(phyloseq)
library(dplyr) 
library(ggplot2)
library(RColorBrewer)
library(colorRamps)
library(data.table)
library(readxl)
library(data.table)
library(microbiome)
library(viridis)
library(forcats)

#create theme for plots
mytheme <- theme_classic() +
  theme(legend.title = element_text(size = 30),
        legend.position ="right", 
        legend.text = element_text(size = 20),
        legend.key.size = unit(1, "cm"),
        legend.key.width = unit(1,"cm"),
        axis.text.x = element_text(face = "bold", size = 18, angle=45, vjust = 0.5),
        axis.text.y = element_text(face = "bold", size = 16),
        axis.title = element_text(face = "bold", size = 16),
        title = element_text(face = "bold", size = 20),
        strip.text = element_text(size = 16))

  level_order <- factor(remainder$Sample, level = c("Control - 0 DAI","Control - 3 DAI", "P. leucotricha - 3 DAI", "V. inaequalis - 3 DAI","Control - 6 DAI", "P. leucotricha - 6 DAI", "V. inaequalis - 6 DAI","Control - 12 DAI", "P. leucotricha - 12 DAI", "V. inaequalis - 12 DAI","Control - 16 DAI", "P. leucotricha - 16 DAI", "V. inaequalis - 16 DAI", "Control - 28 DAI", "P. leucotricha - 28 DAI", "V. inaequalis - 28 DAI","Control - 40 DAI", "P. leucotricha - 40 DAI", "V. inaequalis - 40 DAI","Control - 48 DAI", "V. inaequalis - 48 DAI"
  ))

# Function for colour palette
ColourPalleteMulti <- function(df, group, subgroup){
  # Find how many colour categories to create and the number of colours in each
  categories <- aggregate(as.formula(paste(subgroup, group, sep="~" )), df, function(x) length(unique(x)))
  category.start <- (scales::hue_pal(l = 100)(nrow(categories))) # Set the top of the colour pallete
  category.end  <- (scales::hue_pal(l = 20)(nrow(categories))) # set the bottom
  
  # Build Colour pallette
  colours <- unlist(lapply(1:nrow(categories),
                           function(i){
                             colorRampPalette(colors = c(category.start[i], category.end[i]))(categories[i,2])}))
  return(colours)
}


# Plot for loosely compartment -----------------------------------------------------------------

temp <- ps %>%
  subset_samples( Compartment=="L") %>%
  microbiome::transform("compositional") %>%
  merge_samples("Path_TP") %>%
  transform_sample_counts(function(x) 100 * x/sum(x)) %>%
  tax_glom(taxrank = "F", NArm = FALSE)

dat_taxa <- data.table(psmelt(temp))
dat_taxa$F <- as.character(dat_taxa$F)
medians <- dat_taxa[, mean := mean(Abundance, na.rm = TRUE), by = "F"]
dat_taxa[is.na(dat_taxa)] <- "Unclassified" 
dat_taxa <- dat_taxa[,-c(1,4:23)]
remainder <- dat_taxa[(mean <= 2), F := "Other"]

cat(paste(shQuote(sort(unique(remainder$Sample)), type="cmd"), collapse=", "))
remainder$Sample <- mapvalues(remainder$Sample, from = c("K_0", "K_12", "K_16", "K_28", "K_3", "K_40", "K_48", "K_6", "L_12", "L_16", "L_28", "L_3", "L_40", "L_6", "V_12", "V_16", "V_28", "V_3", "V_40", "V_48", "V_6"), to = c("Control - 0 DAI", "Control - 12 DAI", "Control - 16 DAI", "Control - 28 DAI", "Control - 3 DAI", "Control - 40 DAI", "Control - 48 DAI", "Control - 6 DAI", "P. leucotricha - 12 DAI", "P. leucotricha - 16 DAI", "P. leucotricha - 28 DAI", "P. leucotricha - 3 DAI", "P. leucotricha - 40 DAI", "P. leucotricha - 6 DAI", "V. inaequalis - 12 DAI", "V. inaequalis - 16 DAI", "V. inaequalis - 28 DAI", "V. inaequalis - 3 DAI", "V. inaequalis - 40 DAI", "V. inaequalis - 48 DAI", "V. inaequalis - 6 DAI"))


df <- remainder[(mean <= 2), F := "Other"]
names(df)[names(df) == "P"] <- "Phylum"
names(df)[names(df) == "F"] <- "Family"
df$group <- paste0(df$Phylum, " - ", df$Family, sep = "")
sort(unique(df$group))
cat(paste(shQuote(sort(unique(df$Phylum)), type="cmd"), collapse=", "))

# Choose phyla to keep in plot
phylums <- c("Acidobacteriota", "Actinobacteriota","Bacteroidota", "Bdellovibrionota",
             "Firmicutes", "Proteobacteria", "Unclassified")

df$group[df$Phylum=="Acidobacteriota" & df$Family=='Other'] <- "Acidobacteria - Other"
df$group[df$Phylum=="Actinobacteriota" & df$Family=='Other'] <- "Actinobacteriota - Other"
df$group[df$Phylum=="Bacteroidota" & df$Family=='Other'] <- "Bacteroidota - Other"
df$group[df$Phylum=="Bdellovibrionota" & df$Family=='Other'] <- "Bdellovibrionota - Other"
df$group[df$Phylum=="Firmicutes" & df$Family=='Other'] <- "Firmicutes - Other"
df$group[df$Phylum=="Proteobacteria" & df$Family=='Other'] <- "Proteobacteria - Other"
df$group[df$group=='Unclassified-Other'] <- "Unclassified"
df$group[!df$Phylum %in% phylums] <- "Other Phyla"

df2 <- select(df, Sample, Phylum, Family, Abundance, group) %>%
  mutate(Phylum=factor(Phylum, levels=c(phylums, "Others")),
         Family=fct_reorder(Family, 10*as.integer(Phylum) + grepl("Others", Family)))
df2$group[df2$group=="Proteobacteria-UnknownFamily"] <- "Proteobacteria-Unknown_Family"
df2$group[df2$group=="Proteobacteria-Unknown_Family"] <-"Proteobacteria-Unknown_Family"
df2$group[df2$group=="Acidobacteriota-Acidobacteriaceae(Subgroup1)"] <-"Acidobacteriota Acidobacteriaceae Subgroup 1"

# Assign colors to phyla
sort(unique(df2$group))
colours <- c("chartreuse1","chartreuse4", #Acidobacteriota
             "chocolate4", #Actinobacteriota
             "royalblue1", #Bacteroidota
             "seagreen1", #Bdellovibrionota
             "black", #Firmicutes
             "grey", #Other
             plasma(6), #Proteobacteria
             "white") #Unclassified

# Create plot
ggplot(df2, aes(x=level_order, y=Abundance, fill=group, order=group)) + 
  geom_bar(aes(fill=group), stat="identity", position="stack") + 
  mytheme + theme(axis.text.x = element_text(angle=75, vjust = 0.5)) +
  scale_fill_manual("", values=colours) +
  scale_x_discrete("") +
  theme(
    axis.text.x = element_text(angle=90, vjust=0.5, hjust=1, size=30, colour = "black"), 
    axis.text.y = element_text(size = 22, face = "bold", colour = "black"),
    axis.title.y = element_text(size = 26, face = "bold", colour = "black")
    ) +  # Vertical x-axis tick labels
  scale_y_continuous("Relative Abundance [%]", breaks=seq(0,100,5), limits = c(0, 101), expand = c(0,2)) +
  guides(fill=guide_legend(nrow=length(unique(df2$group)))) + labs(fill = "Family") +
  geom_vline(xintercept = c(1.5,4.5,7.5,10.5,13.5,16.5,19.5), color = "black", size=3)



# Plot for tightly compartment -----------------------------------------------------------------

rm(remainder); remove(df) #remove temporary files

temp <- ps %>%
  subset_samples( Compartment=="T") %>%
  microbiome::transform("compositional") %>%
  merge_samples("Path_TP") %>%
  transform_sample_counts(function(x) 100 * x/sum(x)) %>%
  tax_glom(taxrank = "F", NArm = FALSE)

dat_taxa <- data.table(psmelt(temp))
dat_taxa$F <- as.character(dat_taxa$F)
medians <- dat_taxa[, mean := mean(Abundance, na.rm = TRUE), by = "F"]
dat_taxa[is.na(dat_taxa)] <- "Unclassified" 
dat_taxa <- dat_taxa[,-c(1,4:23)]
remainder <- dat_taxa[(mean <= 2), F := "Other"]

cat(paste(shQuote(sort(unique(remainder$Sample)), type="cmd"), collapse=", "))
remainder$Sample <- mapvalues(remainder$Sample, from = c("K_0", "K_12", "K_16", "K_28", "K_3", "K_40", "K_48", "K_6", "L_12", "L_16", "L_28", "L_3", "L_40", "L_6", "V_12", "V_16", "V_28", "V_3", "V_40", "V_48", "V_6"), to = c("Control - 0 DAI", "Control - 12 DAI", "Control - 16 DAI", "Control - 28 DAI", "Control - 3 DAI", "Control - 40 DAI", "Control - 48 DAI", "Control - 6 DAI", "P. leucotricha - 12 DAI", "P. leucotricha - 16 DAI", "P. leucotricha - 28 DAI", "P. leucotricha - 3 DAI", "P. leucotricha - 40 DAI", "P. leucotricha - 6 DAI", "V. inaequalis - 12 DAI", "V. inaequalis - 16 DAI", "V. inaequalis - 28 DAI", "V. inaequalis - 3 DAI", "V. inaequalis - 40 DAI", "V. inaequalis - 48 DAI", "V. inaequalis - 6 DAI"))


df <- remainder[(mean <= 2), F := "Other"]
names(df)[names(df) == "P"] <- "Phylum"
names(df)[names(df) == "F"] <- "Family"
df$group <- paste0(df$Phylum, " - ", df$Family, sep = "")
sort(unique(df$group))
cat(paste(shQuote(sort(unique(df$Phylum)), type="cmd"), collapse=", "))

# Choose phyla to keep in plot
phylums <- c("Acidobacteriota", "Actinobacteriota","Bacteroidota", "Bdellovibrionota",
             "Firmicutes", "Proteobacteria", "Unclassified")
df$group[df$Phylum=="Acidobacteriota" & df$Family=='Other'] <- "Acidobacteria - Other"
df$group[df$Phylum=="Actinobacteriota" & df$Family=='Other'] <- "Actinobacteriota - Other"
df$group[df$Phylum=="Bacteroidota" & df$Family=='Other'] <- "Bacteroidota - Other"
df$group[df$Phylum=="Bdellovibrionota" & df$Family=='Other'] <- "Bdellovibrionota - Other"
df$group[df$Phylum=="Firmicutes" & df$Family=='Other'] <- "Firmicutes - Other"
df$group[df$Phylum=="Proteobacteria" & df$Family=='Other'] <- "Proteobacteria - Other"
df$group[df$group=='Unclassified-Other'] <- "Unclassified"
df$group[!df$Phylum %in% phylums] <- "Other Phyla"

df2 <- select(df, Sample, Phylum, Family, Abundance, group) %>%
  mutate(Phylum=factor(Phylum, levels=c(phylums, "Others")),
         Family=fct_reorder(Family, 10*as.integer(Phylum) + grepl("Others", Family)))
df2$group[df2$group=="Proteobacteria-UnknownFamily"] <- "Proteobacteria-Unknown_Family"
df2$group[df2$group=="Proteobacteria-Unknown_Family"] <-"Proteobacteria-Unknown_Family"
df2$group[df2$group=="Acidobacteriota-Acidobacteriaceae(Subgroup1)"] <-"Acidobacteriota Acidobacteriaceae Subgroup 1"

# Assign colors to phyla
sort(unique(df2$group))
colours <- c("chartreuse1","chartreuse4", #Acidobacteriota
             "chocolate4", #Actinobacteriota
             "royalblue1", #Bacteroidota
             "seagreen1", #Bdellovibrionota
             "black", #Firmicutes
             "grey", #Other
             plasma(7), #Proteobacteria
             "black") #Unclassified

ggplot(df2, aes(x=level_order, y=Abundance, fill=group, order=group)) + 
  geom_bar(aes(fill=group), stat="identity", position="stack") + 
  mytheme + theme(axis.text.x = element_text(angle=75, vjust = 0.5)) +
  scale_fill_manual("", values=colours) +
  scale_x_discrete("") +
  theme(
    axis.text.x = element_text(angle=90, vjust=0.5, hjust=1, size=30, colour = "black"), 
    axis.text.y = element_text(size = 22, face = "bold", colour = "black"),
    axis.title.y = element_text(size = 26, face = "bold", colour = "black")
  ) +  # Vertical x-axis tick labels
  scale_y_continuous("Relative Abundance [%]", breaks=seq(0,100,5), limits = c(0, 101), expand = c(0,2)) +
  guides(fill=guide_legend(nrow=length(unique(df2$group)))) + labs(fill = "Family") +
  geom_vline(xintercept = c(1.5,4.5,7.5,10.5,13.5,16.5,19.5), color = "black", size=3)
```


--------------------
  
### Differential abundance analysis using ANCOM-BC:
  
##### **Figure S3:**  


```{r, cache=FALSE, warning = FALSE, message = FALSE, error = FALSE}

# Load data, packages and create themes and functions:

load('temporal_trial.RData')
library(ANCOMBC)
library(nlme)
library(tidyverse)
library(compositions)
library(readr)
library(DT)
library(matrixStats)
tax="genus"

dir.create("Plots")
dir.create("stat_results")
dir.create("stat_results/ANCOMBC")
dir.create("stat_results/ANCOMBC/pairwise/")
dir.create("stat_results/ANCOMBC/pairwise/genus")


# Perform ANCOM-BC and save results in table -----------------------------------------------------------------
for (comp in c("L","T")){
  
  var1 <- "TP_cat"
  var2 <- "Treatment"
  pseq <- subset_samples(ps, Compartment==comp) 
  
  # Control analysis --------------------------------------------------------------------
  path="K"
  dir.create(paste0("stat_results/ANCOMBC/pairwise/",tax,"/",path))  
  tmp <- subset_samples(pseq, Control=="YES")
  genus_data <- tax_glom(tmp, taxrank <- rank_names(tmp)[6], NArm = FALSE)
  
  sample_data(genus_data)$TP_cat <- factor(sample_data(genus_data)$TP_cat, levels = 
                                                c("TP_0","TP_3","TP_6","TP_12","TP_16","TP_28","TP_40","TP_48"))
  
  # Run ancombc function
  out = ancombc(phyloseq = genus_data, formula = "TP_cat",
                p_adj_method = "holm", zero_cut = 0.90, lib_cut = 8000,
                group = var1, struc_zero = FALSE, neg_lb = FALSE,
                tol = 1e-5, max_iter = 100, conserve = TRUE,
                alpha = 0.1, global = FALSE)
  res = out$res
  
  #Coefficients
  tab_coef = res$W
  colnames(tab_coef) 
  col_name = c("0:3","0:6","0:12","0:16","0:28","0:40","0:48")
  colnames(tab_coef) = col_name
  source("ancom_bc_rename_variables.R", echo = T, spaced = T)
  tmp <- tab_diff
  tmp2 <- merge(tmp, tab_w, by=0) 
  write_csv(tmp2, file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",path,"/",comp,"-",var2,".csv"))

  # V. inaequalis analysis --------------------------------------------------------------------
  path="V"
  dir.create(paste0("stat_results/ANCOMBC/pairwise/",tax,"/",path))  
  tmp <- subset_samples(pseq, Ventin=="YES")
  genus_data <- tax_glom(tmp, taxrank <- rank_names(tmp)[6], NArm = FALSE)
  
  sample_data(genus_data)$TP_cat <- factor(sample_data(genus_data)$TP_cat, levels = 
                                             c("TP_0","TP_3","TP_6","TP_12","TP_16","TP_28","TP_40","TP_48"))
  
  # Run ancombc function
  out = ancombc(phyloseq = genus_data, formula = "TP_cat",
                p_adj_method = "holm", zero_cut = 0.90, lib_cut = 8000,
                group = var1, struc_zero = FALSE, neg_lb = FALSE,
                tol = 1e-5, max_iter = 100, conserve = TRUE,
                alpha = 0.1, global = FALSE)
  res = out$res
  
  #Coefficients
  tab_coef = res$W
  colnames(tab_coef) 
  col_name = c("0:3","0:6","0:12","0:16","0:28","0:40","0:48")
  colnames(tab_coef) = col_name
  source("ancom_bc_rename_variables.R", echo = T, spaced = T)
  tmp <- tab_diff
  tmp2 <- merge(tmp, tab_w, by=0) 
  write_csv(tmp2, file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",path,"/",comp,"-",var2,".csv"))
  
  # P. leucotricha analysis --------------------------------------------------------------------
  path="L"
  dir.create(paste0("stat_results/ANCOMBC/pairwise/",tax,"/",path))  
  tmp <- subset_samples(pseq, Leuco=="YES")
  genus_data <- tax_glom(tmp, taxrank <- rank_names(tmp)[6], NArm = FALSE)
  
  sample_data(genus_data)$TP_cat <- factor(sample_data(genus_data)$TP_cat, levels = 
                                             c("TP_0","TP_3","TP_6","TP_12","TP_16","TP_28","TP_40"))
  
  # Run ancombc function
  out = ancombc(phyloseq = genus_data, formula = "TP_cat",
                p_adj_method = "holm", zero_cut = 0.90, lib_cut = 8000,
                group = var1, struc_zero = FALSE, neg_lb = FALSE,
                tol = 1e-5, max_iter = 100, conserve = TRUE,
                alpha = 0.1, global = FALSE)
  res = out$res
  
  #Coefficients
  tab_coef = res$W
  colnames(tab_coef) 
  col_name = c("0:3","0:6","0:12","0:16","0:28","0:40")
  colnames(tab_coef) = col_name
  source("ancom_bc_rename_variables.R", echo = T, spaced = T)
  tmp <- tab_diff
  tmp2 <- merge(tmp, tab_w, by=0) 
  write_csv(tmp2, file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",path,"/",comp,"-",var2,".csv"))
  
}


# Combine the three datasets --------------------------------------------------------

for (comp in c("L","T")){
  
  var1 <- "TP_cat"
  var2 <- "Treatment"
  
  path="K";R1 <- read.csv(file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",path,"/",comp,"-",var2,".csv"))
  R1 <- R1[,c(1,7,14)]  
  path="V";R2 <- read.csv(file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",path,"/",comp,"-",var2,".csv"))
  R2 <- R2[,c(1,7,14)]  
  path="L";R3 <- read.csv(file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",path,"/",comp,"-",var2,".csv"))
  R3 <- R3[,c(1,7,13)]  
  tmp <- Reduce(function(x, y) merge(x, y, all=TRUE, by="Row.names"), list(R1,R2,R3))
  rownames(tmp) <- tmp$Row.names; tmp <- tmp[,-1]

  pseq <- subset_samples(ps, Compartment==comp) 
  pseq <- subset_samples(pseq, TP_cat=="TP_40") 
  genus_data <- tax_glom(pseq, taxrank <- rank_names(pseq)[6], NArm = FALSE)
  
  PGroup <- transform_sample_counts(genus_data, function(x)100* x / sum(x))
  OTUg <- otu_table(PGroup)
  AverageD <- as.data.frame(rowMeans(OTUg))
  names(AverageD) <- ("Mean")
  SD <- as.data.frame(rowSds(OTUg),na.rm = T)
  names(SD) <- ("SD")
  tmp_stat <- cbind(AverageD,SD)

  tmp2 <- merge(tmp,tmp_stat, by=0, all=TRUE)
  colnames(tmp2) <- c("Row.names","K=0:40.sig","K=0:40","V=0:40.sig","V=0:40","L=0:40.sig","L=0:40","Mean","SD")
  
  # export results
  write_csv(tmp2, file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",comp,"-",var1,"_results.csv"))
  
}

```

Heatmap plots are created from the ANCOM-BC data

```{r, cache=FALSE, warning = FALSE, message = FALSE}
# Load data, packages and create themes and functions:

load('temporal_trial.RData')
library(pheatmap)
library(RColorBrewer)
library(viridis)
library(dendsort)
library(qiime2R)
library(phyloseq)
library(microbiome)
library(ComplexHeatmap) 
library(circlize)
library(round)
library(dplyr)
library(tidyr)
library(readr)
library(tibble)
tax="genus"
p.colors <- read.table("phyla_colors.tsv", header=T)
p.colors$P <- p.colors$Phylum 

# Treatment plot --------------------------------

tax <- "genus"
var1 <- "TP_cat"
var2 <- "Treatment"

###ANCOM results

for (comp in c("L","T")){
  
  tmp <- read.csv(file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",comp,"-",var1,"_results.csv"))
  
  rownames(tmp) <- tmp$Row.names; tmp$Row.names = NULL
  pseq <- subset_samples(ps, Compartment==comp) 
  genus_data <- tax_glom(pseq, taxrank <- rank_names(pseq)[6], NArm = FALSE)
  
  ttable <- as.data.frame(tax_table(genus_data))
  GTable <- merge(ttable,tmp, by=0)
  rownames(GTable) <- GTable$Row.names; GTable$Row.names = NULL
  head(GTable)
  
  ancom <- GTable[(GTable$Mean>=0.1),] %>% drop_na(P)
  
  ancom <- ancom[,-c(1,7:8)]
  ancom$Phylum <- ancom$P
  ancom <- merge(ancom, p.colors, by="Phylum", all.y = F)
  ancom$G[ancom$G=="uncultured"] <- "Unclassified"
  ancom$G[is.na(ancom$G)] <- "Unclassified"
  ancom <- add_column(ancom, Name = ancom$G, .after = "G")
  ancom <- ancom[(rowSums(ancom[,c(8,10,12)])!=0),] %>% drop_na(Phylum)
  head(ancom)
  ancom <- ancom[,-c(2)]
  colnames(ancom)[1:5] <- c("Phylum","Class","Order","Family","Genus")
  
  tax.clean <- ancom
  colnames(tax.clean)
  
  tax.clean <- rename.ancombc.output(ancom)
  write_csv(tax.clean, file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",comp,"_treatment_subplots_combined.csv"))
  file.copy(from = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",comp,"_treatment_subplots_combined.csv"), to = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",comp,"_treatment_subplots_combined_mod.csv"))
  #redo in excel! -> specify taxa names in "names" column and delete taxonomy columns besides "phylum" and "name"
}
  
for (comp in c("L","T")){
    
  mat.df <- read.csv(file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",comp,"_treatment_subplots_combined_mod.csv"), row.names = "Name")
  str(mat.df)
  
  colnames(mat.df) <- c("Phylum",
                        "K.0.40.sig","C: 0 -> 40 DAI",  
                        "V.0.40.sig","V: 0 -> 40 DAI", 
                        "L.0.40.sig","L: 0 -> 40 DAI", 
                        "Mean","SD","P.color") 
  
  #IMPORTANT: specify ONLY the columns with the differentials
  col.order <- c("C: 0 -> 40 DAI","L: 0 -> 40 DAI","V: 0 -> 40 DAI") 
  
  sig_mat <- mat.df[,c(2,4,6)] #TRUE/FALSE
  mat.diff <- mat.df[,c(3,5,7)] #log fold changes
  min_lc <- min(mat.diff, na.rm = T)
  max_lc <- max(mat.diff, na.rm = T)
  colors <- colorRamp2(c(min_lc, 0, max_lc), c("blue", "white", "red"))
  
  hb1 = rowAnnotation('Mean Abundance\n40 DAI [%]' = anno_barplot(as.vector(mat.df$Mean), gp = gpar(fill = "black"), bar_width = 0.9, extend=0.1, axis_param = list(side = "bottom", labels_rot = 90, gp = gpar(fontsize = 26)), width = unit(10, "cm")), annotation_name_gp = gpar(fontsize = 26, fontface = "bold"),  annotation_name_side = "top", annotation_name_offset = unit(0.3, "cm"), annotation_name_rot = 0)
  
  lgd = Legend(at = c(round(min_lc+min_lc*0.05,digits = 1), 0, round(max_lc+max_lc*0.05, digits = 1)),  col_fun = colors, title = "W-value", title_gp = gpar(fontsize = 26, fontface = "bold"), title_gap = unit(0.5, "cm"), labels_gp = gpar(col = "black", fontsize = 26, fontface = "bold"), title_position = "topcenter", grid_height = unit(3, "cm"), legend_width = unit(10, "cm"), direction = "horizontal") 
  
  ht = Heatmap(mat.diff , name = "logfold change",
               column_gap = unit(5, "mm"),
               row_split = mat.df$Phylum,
               #column_split = c("A","A","B","B","C","C","D","D","E","E","F","F","G"),
               row_gap = unit(5, "mm"), 
               row_names_gp = gpar(fontsize = 32, fontface = "bold", 
                                   col = mat.df$`P-color`),
               na_col = "grey",
               column_order = col.order,
               col = colors,
               rect_gp = gpar(col = "white", lwd = 2), #white spaces in between
               border = TRUE, 
               cluster_rows = FALSE, #remove cluster         
               show_column_dend = FALSE, #remove cluster 
               show_heatmap_legend = FALSE, #remove legend
               row_names_side = "right",
               row_names_max_width = unit(2, "cm"),
               row_names_rot = 0,
               row_names_centered = FALSE,
               row_title_gp = gpar(fontsize = 28),
               row_title_rot = 0,
               column_title = paste0("ANCOM-BC W-value"), 
               column_title_gp = gpar(fill = "black", col = "white", fontsize = 28, fontface = "bold", border = "black"), 
               column_title_side = "top",
               column_names_max_height = unit(6, "cm"),
               column_names_gp = gpar(fontsize = 30, fontface = "bold", col = "black "),
               column_names_rot = 90,
               column_names_centered = TRUE,
               show_parent_dend_line = FALSE, 
               cell_fun = function(j, i, x, y, width, height, fill) {
                 if(sig_mat[i, j] == "TRUE")
                   grid.text("*", x, y, gp = gpar(fontsize = 28, fontface = "bold"))
               }, 
               right_annotation = c(hb1))
  draw(ht, padding = unit(c(5, 1, 1, 26), "cm")) # add space for titles
  draw(lgd, x = unit(80, "cm"), y = unit(5.5, "cm"))
}
```
