---
title: "Mixed Trial"
author: "Maximilian Fernando Becker"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(cache =TRUE)
```


-----------------
  
This document contains all statistical analyses conducted for the manuscript. Note that due to the random iterative nature of some analyses (such as PERMANOVA, CAP & PCA) some of the figure parameters will change slightly during reanalysis, though core results will remain essentially unchanged. 

All data to reproduce analysis can be found here: `https://github.com/mfbeuq/becker_etal_abovegroundpathogen`

-----------------
  
### Load necessary ecological analysis libraries. 

```{r,cache=TRUE, message=FALSE}
library(phyloseq)
library(microbiome)
library(ggord)
library(metagMisc)
library(ggpubr)
library(FSA)
library(knitr)
library(rmarkdown)
library(ape)
library(vegan)
library(philr)
library(compositions)
library(qiime2R)
library(plyr)
library(dplyr)
library(tidyr)
library(PMCMR)
library(tibble)
library(viridis)
library(gridExtra)
library(AcidPlots)
library(grid)
library(colorRamps)
library(rstatix)
library(dunn.test)
library(pairwiseAdonis)
library(dplyr)
library(ANCOMBC)
library(nlme)
library(tidyverse)
library(compositions)
library(readr)
library(DT)
library(matrixStats)
library(pheatmap)
library(RColorBrewer)
library(dendsort)
library(ComplexHeatmap)
library(circlize)
library(round)
library(lme4)
library(emmeans)
set.seed(225)
```

### Contents of this workspace

```{r,cache=TRUE}
load('mixed_trial.RData')
```

* `ps`: phyloseq object including the metadata, and the ASV table, the taxonomy table and the phylogenetic tree, all three created by QIIME2
* `shannon.all`: Shannon diversity index data of the entire root microbiota
* `shannon.L`: Shannon diversity index data of the loosely associated (L) root microbiota sub data set
* `shannon.T`: Shannon diversity index data of the tightly associated (T) root microbiota sub data set
* `RA.dist.L`: DEICODE distance matrix generated by QIIME2 of the loosely associated (L) root microbiota sub data set
* `RA.dist.T`: DEICODE distance matrix generated by QIIME2 of the tightly associated (T) root microbiota sub data set
* `q_meta`: the metadata table 
* `DS`: the disease severity table 
* `deicode_theme`: a theme for generating the plots
* `read_distance`: function for importing a DEICODE distance matrix generated by QIIME2
* `rename.ancombc.output``: a function for renaming taxa in an ANCOM-BC table
* `bd_distances_L_TP3`: PERMDISP DEICODE distances generated by QIIME2 of the loosely associated (L) root microbiota sub data set at TP3
* `bd_distances_T_TP3`: PERMDISP DEICODE distances generated by QIIME2 of the loosely associated (T) root microbiota sub data set at TP3
* `mat.df.L`: formatted ANCOM-BC results table of the loosely associated (L) root microbiota sub data set at TP3
* `mat.df.T`: formatted ANCOM-BC results table of the loosely associated (T) root microbiota sub data set at TP3
* `R2`: Pairwise PERMANOVA R2-values
* `sig`: Pairwise PERMANOVA significance results
* functions and lists for the ordering of graphs


--------------------
  
  
### Disease severity plot and statistics:

##### **Figure 6:**  

```{r, cache=TRUE, warning = FALSE}
#Function to calculate Mean and SD of each Factor
data_summary <- function(data, varname, groupnames){
  require(plyr)
  summary_func <- function(x, col){
    c(mean = mean(x[[col]], na.rm=TRUE),
      sd = sd(x[[col]], na.rm=TRUE))
  }
  data_sum<-ddply(data, groupnames, .fun=summary_func,
                  varname)
  #data_sum <- rename(data_sum, c("mean" = varname))
  return(data_sum)
}

# Extend table
df2 <- DS %>% 
  tidyr::pivot_longer(
    cols = starts_with("SD"), 
    names_to = "Date", 
    values_to = "SD", 
    names_prefix = "SD",
    values_drop_na = TRUE)
df3 <- data_summary(df2, varname="SD", 
                    groupnames=c("Factor", "Date"))
df3$Date <- as.numeric(df3$Date)
df3$Factor <- factor(df3$Factor, levels=c("water_control","water_inoculated","Aliette_inoculated"))
colnames(df3)[3:4] <- c("Mean","SD")

# Plot
figure6A <- ggplot(df3, aes(x=Date, y=Mean, group=Factor, color=Factor)) + 
  geom_errorbar(aes(ymin=Mean-+SD, ymax=Mean+SD), width=.15, size=2, 
                position=position_dodge(0.01)) +
  geom_line(size=2) + geom_point(size=6) + 
  ylab("Disease severity") + 
  scale_x_continuous(limits=c(1, 3.03), name="Sampling timepoint", breaks = c(1, 2, 3)) +
  scale_color_manual(values=c("deepskyblue","darkorchid","firebrick2"), name="Treatment",
                     labels=c("NC Group",  "IU group", "IT group")) + deicode_theme 
print(figure6A)
```


```{r, cache=TRUE, warning = FALSE}
# Statistical analysis 
kruskal.test(SD2 ~ Factor, data = df)
dunn.test(df$SD2 , df$Factor, method="bh", alpha=0.1, list=T)
kruskal.test(SD3 ~ Factor, data = df)
dunn.test(df$SD3 , df$Factor, method="bh", alpha=0.1, list=T)

#Decrease of IT group between TP2 and TP3
IT_group <- df2 %>%
  filter(Factor=="Aliette_inoculated") 
kruskal.test(SD ~ Date, data = IT_group)[3]

#Increase of IU group between TP2 and TP3
IU_group <- df2 %>%
  filter(Factor=="water_inoculated") 
kruskal.test(SD ~ Date, data = IU_group)[3]

``` 


### Alpha diversity statistics and plot:

##### **Figure 7A:**  

```{r, cache=TRUE, warning = FALSE}
shannon <- shannon.all
tmp <- q_meta %>% 
  left_join(shannon) %>%
  filter(!is.na(shannon_entropy)) %>%
  filter(!Compartment=="B") %>%
  filter(!Factor=="")
figure7a <- ggplot(tmp, aes(x=`comp_TP`, y=`shannon_entropy`, fill=`grouped`)) +
geom_boxplot() + deicode_theme +
  ylab("Shannon diversity") + 
  scale_x_discrete(labels=c("L_TP1" = "TP1", "L_TP2" = "TP2", "L_TP3" = "TP3",
                            "T_TP1" = "TP1", "T_TP2" = "TP2", "T_TP3" = "TP3")) + 
  scale_fill_manual(values=c("firebrick2","firebrick2","firebrick2","deepskyblue","darkorchid","darkorchid","darkorchid"), name="Treatment", labels=c("","","IT group", "NC group","IU group","","")) +
  theme(axis.title.x=element_blank(), axis.title.y=element_text(size=30), axis.text.y=element_text(size=24), axis.text.x=element_text(size=20))
print(figure7a)
```

The following does the statistical analysis for the Shannon index values of the L-compartment

```{r, cache=TRUE, warning = FALSE}
Model <- lm(Shannon ~ Factor*TP_cat, data = ps.meta.L)
anova(Model)
```

The following does the statistical analysis for the Shannon index values of the T-compartment

```{r, cache=TRUE, warning = FALSE}
Model <- lm(Shannon ~ Factor*TP_cat, data = ps.meta.T)
anova(Model)
```

--------------------
  
### Beta Diversity:
  
  
##### **Loosely associated root microbiome:**  

First, the distance matrix calculated by QIIME2 needs to be merged with the metadata file:

```{r, cache=TRUE}
RA.dist <- RA.dist.L
tmp_order <- colnames(RA.dist)
row.names(q_meta) <- q_meta$SampleID
de <- merge(q_meta, RA.dist, by=0, all=F, all.y=F)
row.names(de) <- de$SampleID
de <- de[ order(match(row.names(de), tmp_order)), ]
RA.dist <- RA.dist[ order(match(row.names(RA.dist), tmp_order)), ]
```

Then, PERMANOVA is calculated on the distance matrix
```{r, cache=TRUE}
adonis2(formula = RA.dist ~ de$Factor * de$TP_cat, permutations = 999) 
```

Pairwise PERMANOVA is calculated for the grouped factor (combination of timepoint and treatment) and the timepoint
```{r, cache=FALSE, warning = FALSE, message = FALSE, error = FALSE}
pairwise.adonis2(RA.dist ~ grouped, data=de)
pairwise.adonis2(RA.dist ~ TP_cat, data=de)
```

The CAP plot for the L-compartment by using dbrda on the the DEICODE distance matrix
```{r, cache=FALSE, warning = FALSE, message = FALSE, error = FALSE}
capscale <- capscale(formula = RA.dist ~ de$Factor * de$TP_cat * de$SD_final)
x <- as.data.frame(scores(capscale, display = "sites")) 
tmp <- merge(q_meta, x, by=0) %>% 
  mutate(grouped=factor(grouped, levels=group_order)) 
row.names(tmp) <- tmp$SampleID
p1 <- ggplot(tmp, aes(x= CAP1, y= CAP2, colour = grouped, shape = TP_cat, size = SD_final)) +
  geom_point(alpha=0.9) +
  scale_colour_manual(name="Treatment", values= col_graphs,
                      labels = c("NC Group", 
                                 "IU group - TP1", "IU group - TP2", "IU group - TP3",
                                 "IT group - TP1", "IT group - TP2", "IT group - TP3")) +
  scale_shape_manual(values=c(16,18,15), name="Timepoint") +
  scale_size_continuous(name="Disease severity",range = c(8,16)) +
  geom_hline(yintercept = 0, linetype="dotted") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  xlab(paste("CAP1 - ",format(round(100*summary(capscale)$cont$importance[2,1],digits = 1), nsmall = 1),"%")) +
  ylab(paste("CAP2 - ",format(round(100*summary(capscale)$cont$importance[2,2],digits = 1), nsmall = 1),"%")) +
  theme(plot.title = element_text(hjust = 0.5))  + deicode_theme + theme(legend.key.size = unit(0.8, "cm")) + 
  guides(colour = guide_legend(order = 1, override.aes = list(colour = col_graphs, shape = c(15), size=12)),
         shape = guide_legend(order = 2, override.aes = list(shape = c(16,18,15), size=12)),
         size = guide_legend(order = 3, override.aes = list(shape = c(15))))
```


##### **Tightly associated root microbiome:**  

First, the distance matrix calculated by QIIME2 needs to be merged with the metadata file:

```{r, cache=TRUE}
RA.dist <- RA.dist.T
tmp_order <- colnames(RA.dist)
row.names(q_meta) <- q_meta$SampleID
de <- merge(q_meta, RA.dist, by=0, all=F, all.y=F)
row.names(de) <- de$SampleID
de <- de[ order(match(row.names(de), tmp_order)), ]
RA.dist <- RA.dist[ order(match(row.names(RA.dist), tmp_order)), ]
```

Then, PERMANOVA is calculated on the distance matrix
```{r, cache=TRUE}
adonis2(formula = RA.dist ~ de$Factor * de$TP_cat, permutations = 999) 
```

Pairwise PERMANOVA is calculated for the grouped factor (combination of timepoint and treatment) and the timepoint
```{r, cache=FALSE, warning = FALSE, message = FALSE, error = FALSE}
pairwise.adonis2(RA.dist ~ grouped, data=de)
pairwise.adonis2(RA.dist ~ TP_cat, data=de)
```


The PCoA plot for the T-compartment by using the DEICODE ordination
```{r, cache=FALSE, warning = FALSE, message = FALSE, error = FALSE}
capscale <- capscale(formula = RA.dist ~ de$Factor * de$TP_cat * de$SD_final)
x <- as.data.frame(scores(capscale, display = "sites")) 
tmp <- merge(q_meta, x, by=0) %>% 
  mutate(grouped=factor(grouped, levels=group_order)) 
row.names(tmp) <- tmp$SampleID
p2 <- ggplot(tmp, aes(x= CAP1, y= CAP2, colour = grouped, shape = TP_cat, size = SD_final)) +
  geom_point(alpha=0.9) +
  scale_colour_manual(name="Treatment", values= col_graphs,
                      labels = c("NC Group", 
                                 "IU group - TP1", "IU group - TP2", "IU group - TP3",
                                 "IT group - TP1", "IT group - TP2", "IT group - TP3")) +
  scale_shape_manual(values=c(16,18,15), name="Timepoint") +
  scale_size_continuous(name="Disease severity",range = c(8,16)) +
  geom_hline(yintercept = 0, linetype="dotted") + 
  geom_vline(xintercept = 0, linetype="dotted") +
  xlab(paste("CAP1 - ",format(round(100*summary(capscale)$cont$importance[2,1],digits = 1), nsmall = 1),"%")) +
  ylab(paste("CAP2 - ",format(round(100*summary(capscale)$cont$importance[2,2],digits = 1), nsmall = 1),"%")) +
  theme(plot.title = element_text(hjust = 0.5))  + deicode_theme + theme(legend.key.size = unit(0.8, "cm")) + 
  guides(colour = guide_legend(order = 1, override.aes = list(colour = col_graphs, shape = c(15), size=12)),
         shape = guide_legend(order = 2, override.aes = list(shape = c(16,18,15), size=12)),
         size = guide_legend(order = 3, override.aes = list(shape = c(15))))
```


##### **Figure 7C:**  
```{r, cache=FALSE, warning = FALSE, message = FALSE, error = FALSE}

```


##### **Figure 9: Combine both plots:**  

```{r, cache=FALSE, warning = FALSE, message = FALSE, error = FALSE}
legend2 <- cowplot::get_legend(p1)
d1 <- ggpar(p1, legend = "", legend.title = "", font.x = c(22, "bold"), font.y = c(22, "bold"), font.tickslab = c(20, color="black")) +
  annotate("text", y= max(p1$data$CAP2), x =max(p1$data$CAP1), label="(A)", color="black", size=12, hjust=0.5, vjust=0.6) 
d2 <- ggpar(p2, legend = "", legend.title = "", font.x = c(22, "bold"), font.y = c(22, "bold"), font.tickslab = c(20, color="black")) +
  annotate("text", y= max(p2$data$CAP2), x =max(p2$data$CAP1), label="(B)", color="black", size=12, hjust=0.5, vjust=0.6) 
figure9 <- ggarrange(d1, NULL,
                    NULL, legend2,
                    d2, NULL,
                    widths = c(2.5,1), heights = c(1.5,0.05,1.5),
                    ncol = 2, nrow = 3)
```



##### **Figure 8:**  

```{r, cache=FALSE, warning = FALSE, message = FALSE, error = FALSE}
# calculate pairwise PERMANOVA 
for (comp in c("L", "T")){
  RA.dist <- RA.dist.L
  # or
  RA.dist <- RA.dist.T
  tmp_order <- colnames(RA.dist)
  row.names(q_meta) <- q_meta$SampleID
  de <- merge(q_meta, RA.dist, by=0, all=F, all.y=F)
  row.names(de) <- de$SampleID
  de <- de[ order(match(row.names(de), tmp_order)), ]
  RA.dist <- RA.dist[ order(match(row.names(RA.dist), tmp_order)), ]
  pairwise.adonis(x=RA.dist, factors=de[,"grouped"], perm = 999, p.adjust.m='BH')
}

# format results and enter them into a excel sheet (resulting files are attached in workspace)

sig <- sig
rownames(sig) <- sig$Timepoint; sig$Timepoint = NULL 
sig <- as.matrix(sig)
colnames(sig) <- tm_list
rownames(sig) <- tm_list

R2 <- R2
rownames(R2) <- R2$Timepoint; R2$Timepoint = NULL 
R2 <- as.matrix(R2)
colnames(R2) <- tm_list
rownames(R2) <- tm_list

sig.star <- sig
sig.star[is.na(sig.star)] <- ""

colors <- colorRamp2(c(0, 0.5, 1), c("beige","red3", "darkorchid4"), space = "LAB")

lgd = Legend(at = c(0, 1),  col_fun = colors, 
             title = expression(paste("R"^"2", " value")), title_gp = gpar(fontsize = 26, fontface = "bold"), title_gap = unit(0.5, "cm"),
             labels_gp = gpar(col = "black", fontsize = 26, fontface = "bold"), title_position = "topcenter", 
             grid_height = unit(3, "cm"), legend_width = unit(8, "cm"), direction = "horizontal") 

ht = Heatmap(R2, name = "logfold change",
             column_gap = unit(5, "mm"),
             row_gap = unit(5, "mm"), 
             row_names_gp = gpar(fontsize = 32, fontface = "bold"),
             na_col = "grey30",
             col = colors,
             rect_gp = gpar(col = "white", lwd = 2), #white spaces in between
             border = TRUE, 
             cluster_rows = FALSE, #remove cluster
             cluster_columns = FALSE,
             show_column_dend = FALSE, #remove cluster 
             show_heatmap_legend = FALSE, #remove legend
             row_names_side = "right",
             row_names_max_width = unit(2, "cm"),
             row_names_rot = 0,
             row_names_centered = TRUE,
             row_title_gp = gpar(fontsize = 28),
             row_title_rot = 0,
             column_title = expression(paste("PERMANOVA R"^"2", " value")),
             column_title_gp = gpar(fill = "black", col = "white", fontsize = 28, fontface = "bold", border = "black"), 
             column_title_side = "top",
             column_names_max_height = unit(6, "cm"),
             column_names_gp = gpar(fontsize = 30, fontface = "bold", col = "black"),
             column_names_rot = 90,
             column_names_centered = TRUE,
             show_parent_dend_line = FALSE,
             cell_fun = function(j, i, x, y, width, height, fill) {
               if(sig.star[i, j] == "**")
                 grid.text("**", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
               if(sig.star[i, j] == "*")
                 grid.text("*", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
               if(sig.star[i, j] == ".")
                 grid.text(".", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
               if(sig.star[i, j] == "0")
                 grid.text("", x, y, gp = gpar(fontsize = 32, fontface = "bold"))
             }
)

draw(ht, padding = unit(c(4, 1, 1, 8), "cm")) # add space for titles
draw(lgd, x = unit(30, "cm"), y = unit(4, "cm"))
```



##### **Figure S4:**  

The distances are taken from the QIIME2 generated permdisp table (attached in workspace) 

```{r, cache=FALSE, warning = FALSE, message = FALSE, error = FALSE}

# Loosely -----------------------------------------------------------------
tmp <- bd_distances_L_TP3
tmp[tmp == "Aliette_inoculated"] <- 'IT group'
tmp[tmp == "water_inoculated"] <- 'IU group'
tmp[tmp == "water_control"] <- 'NC group'


tmp2 <- tmp %>% 
  filter( tmp$Group1=="NC group") %>%
  mutate( comparison=paste(Group1, Group2))
ggqqplot(tmp2$Distance); shapiro.test(tmp2$Distance)

bd1 <- tmp2 %>%
  mutate( Group2 = factor(Group2, levels = c("NC group", "IT group", "IU group"))) %>%
  ggplot( aes(x = Group2, y = Distance, fill = Group2, color = Group2)) +
  geom_boxplot(lwd=2, notch=T) +
  ylab("Distance") +
  xlab("") + 
  scale_color_manual(name="Treatment", values = c("black","orangered1","steelblue3"), aesthetics = c("fill")) +
  scale_color_manual(name="", values = c("black","black","black"), aesthetics = c("color")) +
  stat_compare_means(comparisons = my_comparisons, label.y = c(4.5, 5, 5.5), symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "ns")))+
  stat_compare_means(label.y = 6.5, size = 12) +
  deicode_theme + theme(axis.text.y = element_text(face = "bold", size = 20, colour = "black"))
bd1$layers[[2]]$aes_params$textsize <- 16
bd1


# Tightly -----------------------------------------------------------------
tmp <- bd_distances_T_TP3
tmp[tmp == "Aliette_inoculated"] <- 'IT group'
tmp[tmp == "water_inoculated"] <- 'IU group'
tmp[tmp == "water_control"] <- 'NC group'

tmp2 <- tmp %>% 
  filter( tmp$Group1=="NC group") %>%
  mutate( comparison=paste(Group1, Group2))
ggqqplot(tmp2$Distance); shapiro.test(tmp2$Distance)

bd2 <- tmp2 %>%
  mutate( Group2 = factor(Group2, levels = c("NC group", "IT group", "IU group"))) %>%
  ggplot( aes(x = Group2, y = Distance, fill = Group2, color = Group2)) +
  geom_boxplot(lwd=2, notch=T) +
  ylab("Distance") +
  xlab("") + 
  scale_color_manual(name="Treatment", values = c("black","orangered1","steelblue3"), aesthetics = c("fill")) +
  scale_color_manual(name="", values = c("black","black","black"), aesthetics = c("color")) +
  stat_compare_means(comparisons = my_comparisons, label.y = c(4.5, 5, 5.5), symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "ns")))+
  stat_compare_means(label.y = 6.5, size = 12) +
  deicode_theme + theme(axis.text.y = element_text(face = "bold", size = 20, colour = "black"))
bd2$layers[[2]]$aes_params$textsize <- 16
bd2

```



### Differential abundance analysis using ANCOM-BC:
  
##### **Figure 10:**  


```{r, cache=FALSE, warning = FALSE, message = FALSE, error = FALSE}
load('mixed_trial.RData')
library(ANCOMBC)
library(nlme)
library(tidyverse)
library(compositions)
library(readr)
library(DT)
library(matrixStats)
tax="genus"

dir.create(paste0("stat_results/ANCOMBC/"))  
dir.create(paste0("stat_results/ANCOMBC/pairwise"))  
dir.create(paste0("stat_results/ANCOMBC/pairwise/",tax))  

# Treatment: calculate -----------------------------------------------------------------
for (comp in c("L","T")){
  
  pseq <- subset_samples(ps, Compartment==comp) 
  pseq <- tax_glom(pseq, taxrank = rank_names(pseq)[6], NArm = FALSE) 
  var1 <- "Factor"

  # TP3: Control --------------------------------------------------------------------
  rs="TP3"
  dir.create(paste0("stat_results/ANCOMBC/pairwise/",tax,"/",rs))  
  pseq <- subset_samples(ps, Compartment==comp) 
  var2 <- "water_control"
  
  genus_data <- subset_samples(pseq, TP_cat=="TP3")
  sample_data(genus_data)$Factor <- factor(sample_data(genus_data)$Factor, levels = 
                                             c("water_control","water_inoculated","Aliette_inoculated"))
  
  # Run ancombc function
  out = ancombc(phyloseq = genus_data, formula = "Factor",
                p_adj_method = "holm", zero_cut = 0.90, lib_cut = 8000,
                group = var1, struc_zero = FALSE, neg_lb = FALSE,
                tol = 1e-5, max_iter = 100, conserve = TRUE,
                alpha = 0.1, global = FALSE)
  res = out$res
  
  #Coefficients
  tab_coef = res$W
  colnames(tab_coef) 
  col_name = c("Control:Inoculated", "Control:Aliette")
  colnames(tab_coef) = col_name
  
  source("ancom_bc_rename_variables.R", echo = T, spaced = T)
  
  tmp <- tab_diff
  tmp2 <- merge(tmp, tab_w, by=0) 
  write_csv(tmp2, file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",rs,"/",comp,"-",var2,".csv"))
  
  
  # TP3: Inoculated --------------------------------------------------------------------
  rs="TP3"
  dir.create(paste0("stat_results/ANCOMBC/pairwise/",tax,"/",rs))  
  pseq <- subset_samples(ps, Compartment==comp) 
  var2 <- "water_inoculated"
  
  genus_data <- subset_samples(pseq, TP_cat=="TP3")
  sample_data(genus_data)$Factor <- factor(sample_data(genus_data)$Factor, levels = 
                                             c("water_inoculated","water_control","Aliette_inoculated"))
  
  # Run ancombc function
  out = ancombc(phyloseq = genus_data, formula = "Factor",
                p_adj_method = "holm", zero_cut = 0.90, lib_cut = 8000,
                group = var1, struc_zero = FALSE, neg_lb = FALSE,
                tol = 1e-5, max_iter = 100, conserve = TRUE,
                alpha = 0.1, global = FALSE)
  res = out$res
  
  #Coefficients
  tab_coef = res$W
  colnames(tab_coef) 
  col_name = c("Inoculated:Control", "Inoculated:Aliette")
  colnames(tab_coef) = col_name
  
  source("ancom_bc_rename_variables.R", echo = T, spaced = T)
  
  tmp <- tab_diff
  tmp2 <- merge(tmp, tab_w, by=0) 
  write_csv(tmp2, file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",rs,"/",comp,"-",var2,".csv"))
    
}

# Treatment: combine datasets --------------------------------------------------------

for (comp in c("L","T")){
  
  pseq <- subset_samples(ps, Compartment==comp) 
  genus_data <- tax_glom(pseq, taxrank = rank_names(pseq)[6], NArm = FALSE) 
  
  rs="TP3"; var2="water_control";
  R3 <- read.csv(file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",rs,"/",comp,"-",var2,".csv"))
  rs="TP3"; var2="water_inoculated";
  R4 <- read.csv(file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",rs,"/",comp,"-",var2,".csv"))[,c(1,3,5)]
  
  tmp <- Reduce(function(x, y) merge(x, y, all=TRUE, by="Row.names"), list(R3,R4))
  rownames(tmp) <- tmp$Row.names; tmp <- tmp[,-1]
  
  rs <- "TP3"
  pseq <- subset_samples(genus_data, TP_cat==rs)
  PGroup <- transform_sample_counts(pseq, function(x)100* x / sum(x))
  OTUg <- otu_table(PGroup)
  AverageD <- as.data.frame(rowMeans(OTUg)); names(AverageD) <- paste0(rs,".Mean")
  SD <- as.data.frame(rowSds(OTUg),na.rm = T); names(SD) <- paste0(rs,".SD")
  tmp.stat.TP3 <- cbind(AverageD,SD)
  mean.sd <- merge(tmp.stat.TP1, tmp.stat.TP2, by=0) 
  rownames(mean.sd) <- mean.sd$Row.names; mean.sd <- mean.sd[,-1]
  mean.sd <- merge(mean.sd, tmp.stat.TP3, by=0) 
  rownames(mean.sd) <- mean.sd$Row.names; mean.sd <- mean.sd[,-1]
  head(mean.sd)
  
  tmp <- merge(tmp, mean.sd, by=0) 
  write_csv(tmp, file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",comp,"_results.csv"))
    
}
```

Heatmap plots are created from the ANCOM-BC data

```{r, cache=FALSE, warning = FALSE, message = FALSE, error = FALSE}
# Load data, packages and create themes and functions:
load('mixed_trial.RData')
library(ANCOMBC)
library(nlme)
library(tidyverse)
library(compositions)
library(readr)
library(DT)
library(matrixStats)
library(ComplexHeatmap)
tax="genus"
dir.create(paste0("Plots/ANCOM-BC/"))  

# choose Compartment
mat.df <- mat.df.L
mat.df <- mat.df.T

for (comp in c("L","T")){
  
  mat.df.L <- read.csv(file = paste0("stat_results/ANCOMBC/pairwise/",tax,"/",comp,"_subplots_combined_mod.csv"), row.names = "Name")
  str(mat.df)
  
  colnames(mat.df) <- c("Phylum",                
                        "TP1: Inoculated.Aliette.sig", "TP1: IU -> IT", 
                        "TP2: Inoculated.Aliette.sig", "TP2: IU -> IT", 
                        "TP3: Control.Inoculated.sig", "TP3: Control.Aliette.sig", 
                        "TP3: NC -> IU","TP3: NC -> IT", 
                        "TP3: Inoculated.Aliette.sig", "TP3: IU -> IT",
                        "TP1.Mean","TP1.SD","TP2.Mean","TP2.SD","TP3.Mean","TP3.SD","P.color")   
  
  col.order <- c("TP3: NC -> IU","TP3: NC -> IT", "TP3: IU -> IT") 
  #add the names of your factors in your desired order
  #IMPORTANT: specify ONLY the columns with the correct differentials (values)
  mat.diff <- as.matrix(mat.df[,c(8,9,11)])
  #IMPORTANT: specify ONLY the columns with the correct TRUE/FALSE values
  sig_mat <- as.matrix(mat.df[,c(6,7,10)])
  min_lc <- min(mat.diff, na.rm = T)
  max_lc <- max(mat.diff, na.rm = T)
  colors <- colorRamp2(c(min_lc, 0, max_lc), c("blue", "white", "red"))
  
  hb = rowAnnotation('Mean Abundance [%]' = anno_barplot(as.vector(mat.df$TP3.Mean), gp = gpar(fill = "black"), bar_width = 0.9, extend=0.1,
                                                         axis_param = list(side = "bottom", labels_rot = 90, gp = gpar(fontsize = 26)), 
                                                         width = unit(10, "cm")), annotation_name_gp = gpar(fontsize = 26, fontface = "bold"),
                     annotation_name_side = "top", annotation_name_offset = unit(0.3, "cm"), annotation_name_rot = 0)
  
  lgd = Legend(at = c(round(min_lc+min_lc*0.05,digits = 1), 0, round(max_lc+max_lc*0.05, digits = 1)),  col_fun = colors, 
               title = "W-value", title_gp = gpar(fontsize = 26, fontface = "bold"), title_gap = unit(0.5, "cm"),
               labels_gp = gpar(col = "black", fontsize = 26, fontface = "bold"), title_position = "topcenter", 
               grid_height = unit(3, "cm"), legend_width = unit(10, "cm"), direction = "horizontal") 
  
  ht = Heatmap(mat.diff , name = "logfold change",
               column_gap = unit(5, "mm"),
               row_split = mat.df$Phylum,
               row_gap = unit(5, "mm"), 
               row_names_gp = gpar(fontsize = 32, fontface = "bold", 
                                   col = mat.df$`P-color`),
               na_col = "grey",
               column_order = col.order,
               col = colors,
               rect_gp = gpar(col = "white", lwd = 2), #white spaces in between
               border = TRUE, 
               cluster_rows = FALSE, #remove cluster         
               show_column_dend = FALSE, #remove cluster 
               show_heatmap_legend = FALSE, #r emove legend
               row_names_side = "right",
               row_names_max_width = unit(2, "cm"),
               row_names_rot = 0,
               row_names_centered = FALSE,
               row_title_gp = gpar(fontsize = 28),
               row_title_rot = 0,
               column_title = paste0("ANCOM-BC W-value"), 
               column_title_gp = gpar(fill = "black", col = "white", fontsize = 28, fontface = "bold", border = "black"), 
               column_title_side = "top",
               column_names_max_height = unit(6, "cm"),
               column_names_gp = gpar(fontsize = 30, fontface = "bold", col = "black"),
               column_names_rot = 90,
               column_names_centered = TRUE,
               show_parent_dend_line = FALSE, 
               #this functions adds stars when the change in this field is significant (q-value)
               cell_fun = function(j, i, x, y, width, height, fill) {
                 if(sig_mat[i, j] == "TRUE")
                   grid.text("*", x, y, gp = gpar(fontsize = 28, fontface = "bold"))
               }, 
               right_annotation = c(hb))
  png(paste0("Plots/ANCOM-BC/",comp,"-",tax,"_heatmap_differentials_TP3.png"), width = 2000, height = 500+45*(nrow(mat.diff)))
  draw(ht, padding = unit(c(8, 1, 1, 26), "cm")) # add space for titles (you might have to adjust those values)
  draw(lgd, x = unit(45, "cm"), y = unit(10, "cm"))
  dev.off()
}
```


--------------------
  


### Stacked barplot:
  
##### **Figure S3:**  


```{r, cache=FALSE, warning = FALSE, message = FALSE, error = FALSE}

# Load data, packages and create themes and functions:
load('mixed_trial.RData')

library(plyr)
library(phyloseq)
library(dplyr) 
library(ggplot2)
library(RColorBrewer)
library(colorRamps)
library(data.table)
library(readxl)
library(data.table)
library(microbiome)
library(viridis)
library(forcats)

#create theme and functions for plots
mytheme <- theme_classic() +
  theme(legend.title = element_text(size = 30),
        legend.position ="right", 
        legend.text = element_text(size = 20),
        legend.key.size = unit(1, "cm"),
        legend.key.width = unit(1,"cm"),
        axis.text.x = element_text(face = "bold", size = 18, angle=45, vjust = 0.5),
        axis.text.y = element_text(face = "bold", size = 16),
        axis.title = element_text(face = "bold", size = 16),
        title = element_text(face = "bold", size = 20),
        strip.text = element_text(size = 16))

ColourPalleteMulti <- function(df, group, subgroup){
  # Find how many colour categories to create and the number of colours in each
  categories <- aggregate(as.formula(paste(subgroup, group, sep="~" )), df, function(x) length(unique(x)))
  category.start <- (scales::hue_pal(l = 100)(nrow(categories))) # Set the top of the colour pallete
  category.end  <- (scales::hue_pal(l = 20)(nrow(categories))) # set the bottom
  
  # Build Colour pallette
  colours <- unlist(lapply(1:nrow(categories),
                           function(i){
                             colorRampPalette(colors = c(category.start[i], category.end[i]))(categories[i,2])}))
  return(colours)
}

level_order <- factor(remainder$Sample, level = c("TP1: IU", "TP1: IT", 
                                                  "TP2: IU", "TP2: IT", 
                                                  "TP3: NC", "TP3: IU", "TP3: IT"
                                                  ))


# Loosely -----------------------------------------------------------------
temp <- ps %>%
  subset_samples( Compartment=="L") %>%
  microbiome::transform("compositional") %>%
  merge_samples("grouped") %>%
  transform_sample_counts(function(x) 100 * x/sum(x)) %>%
  tax_glom(taxrank = "F", NArm = FALSE)

dat_taxa <- data.table(psmelt(temp))
dat_taxa$F <- as.character(dat_taxa$F)
medians <- dat_taxa[, mean := mean(Abundance, na.rm = TRUE), by = "F"]
dat_taxa[is.na(dat_taxa)] <- "Unclassified" 
dat_taxa <- dat_taxa[,-c(1,4:27)]
remainder <- dat_taxa[(mean <= 2), F := "Other"]

cat(paste(shQuote(sort(unique(remainder$Sample)), type="cmd"), collapse=", "))
remainder$Sample <- mapvalues(remainder$Sample, from = c("Aliette_inoculated_TP1", "Aliette_inoculated_TP2", "Aliette_inoculated_TP3", "water_control_TP3", "water_inoculated_TP1", "water_inoculated_TP2", "water_inoculated_TP3"),
                            to = c("TP1: IT", "TP2: IT", "TP3: IT", "TP3: NC", "TP1: IU", "TP2: IU", "TP3: IU"))

remove(df)
remove(tmp1)
remove(tmp2)
tmp1 <- remainder
tmp2 <- tmp1[(mean <= 2), F := "Other"]


df <- tmp2
names(df)[names(df) == "P"] <- "Phylum"
names(df)[names(df) == "F"] <- "Family"
df$group <- paste0(df$Phylum, " - ", df$Family, sep = "")
sort(unique(df$group))
cat(paste(shQuote(sort(unique(df$Phylum)), type="cmd"), collapse=", "))

# Assign phyla to keep
phylums <- c("Acidobacteriota", "Actinobacteriota","Bacteroidota", "Bdellovibrionota",
             "Firmicutes", "Proteobacteria", "Unclassified")

df$group[df$Phylum=="Acidobacteriota" & df$Family=='Other'] <- "Acidobacteria - Other"
df$group[df$Phylum=="Actinobacteriota" & df$Family=='Other'] <- "Actinobacteriota - Other"
df$group[df$Phylum=="Bacteroidota" & df$Family=='Other'] <- "Bacteroidota - Other"
df$group[df$Phylum=="Bdellovibrionota" & df$Family=='Other'] <- "Bdellovibrionota - Other"
df$group[df$Phylum=="Firmicutes" & df$Family=='Other'] <- "Firmicutes - Other"
df$group[df$Phylum=="Proteobacteria" & df$Family=='Other'] <- "Proteobacteria - Other"
df$group[df$group=='Unclassified-Other'] <- "Unclassified"
df$group[!df$Phylum %in% phylums] <- "Other Phyla"


df2 <- select(df, Sample, Phylum, Family, Abundance, group) %>%
  mutate(Phylum=factor(Phylum, levels=c(phylums, "Others")),
         Family=fct_reorder(Family, 10*as.integer(Phylum) + grepl("Others", Family)))
df2$group[df2$group=="Proteobacteria-UnknownFamily"] <- "Proteobacteria-Unknown_Family"
df2$group[df2$group=="Proteobacteria-Unknown_Family"] <-"Proteobacteria-Unknown_Family"
df2$group[df2$group=="Acidobacteriota-Acidobacteriaceae(Subgroup1)"] <-"Acidobacteriota Acidobacteriaceae Subgroup 1"

sort(unique(df2$group))
colours <- c("chartreuse1","chartreuse4", #Acidobacteriota
             "chocolate4", #Actinobacteriota
             "royalblue1", #Bacteroidota
             "seagreen1", #Bdellovibrionota
             "black", #Firmicutes
             "grey", #Other
             plasma(8), #Proteobacteria
             "white") #Unclassified

# Plot
ggplot(df2, aes(x=level_order, y=Abundance, fill=group, order=group)) + 
  geom_bar(aes(fill=group), stat="identity", position="stack") + 
  mytheme + theme(axis.text.x = element_text(angle=75, vjust = 0.5)) +
  scale_fill_manual("", values=colours) +
  scale_x_discrete("") +
  theme(
    axis.text.x = element_text(angle=90, vjust=0.5, hjust=1, size=30, colour = "black"), 
    axis.text.y = element_text(size = 22, face = "bold", colour = "black"),
    axis.title.y = element_text(size = 26, face = "bold", colour = "black")
    ) +  # Vertical x-axis tick labels
  scale_y_continuous("Relative Abundance [%]", breaks=seq(0,100,5), limits = c(0, 101), expand = c(0,2)) +
  guides(fill=guide_legend(nrow=length(unique(df2$group)))) + labs(fill = "Family") +
  geom_vline(xintercept = c(2.5,4.5), color = "black", size=3)


# Tightly -----------------------------------------------------------------
temp <- ps %>%
  subset_samples( Compartment=="T") %>%
  microbiome::transform("compositional") %>%
  merge_samples("grouped") %>%
  transform_sample_counts(function(x) 100 * x/sum(x)) %>%
  tax_glom(taxrank = "F", NArm = FALSE)

dat_taxa <- data.table(psmelt(temp))
dat_taxa$F <- as.character(dat_taxa$F)
medians <- dat_taxa[, mean := mean(Abundance, na.rm = TRUE), by = "F"]
dat_taxa[is.na(dat_taxa)] <- "Unclassified" 
dat_taxa <- dat_taxa[,-c(1,4:27)]
remainder <- dat_taxa[(mean <= 2), F := "Other"]


cat(paste(shQuote(sort(unique(remainder$Sample)), type="cmd"), collapse=", "))
remainder$Sample <- mapvalues(remainder$Sample, from = c("Aliette_inoculated_TP1", "Aliette_inoculated_TP2", "Aliette_inoculated_TP3", "water_control_TP3", "water_inoculated_TP1", "water_inoculated_TP2", "water_inoculated_TP3"),
                              to = c("TP1: IT", "TP2: IT", "TP3: IT", "TP3: NC", "TP1: IU", "TP2: IU", "TP3: IU"))

remove(df)
remove(tmp1)
remove(tmp2)
tmp1 <- remainder
tmp2 <- tmp1[(mean <= 2), F := "Other"]


df <- tmp2
names(df)[names(df) == "P"] <- "Phylum"
names(df)[names(df) == "F"] <- "Family"
df$group <- paste0(df$Phylum, " - ", df$Family, sep = "")
sort(unique(df$group))
cat(paste(shQuote(sort(unique(df$Phylum)), type="cmd"), collapse=", "))

#Assign phyla to keep
phylums <- c("Acidobacteriota", "Actinobacteriota","Bacteroidota", "Bdellovibrionota",
             "Firmicutes", "Proteobacteria", "Unclassified")

df$group[df$Phylum=="Acidobacteriota" & df$Family=='Other'] <- "Acidobacteria - Other"
df$group[df$Phylum=="Actinobacteriota" & df$Family=='Other'] <- "Actinobacteriota - Other"
df$group[df$Phylum=="Bacteroidota" & df$Family=='Other'] <- "Bacteroidota - Other"
df$group[df$Phylum=="Bdellovibrionota" & df$Family=='Other'] <- "Bdellovibrionota - Other"
df$group[df$Phylum=="Firmicutes" & df$Family=='Other'] <- "Firmicutes - Other"
df$group[df$Phylum=="Proteobacteria" & df$Family=='Other'] <- "Proteobacteria - Other"
df$group[df$group=='Unclassified-Other'] <- "Unclassified"
df$group[!df$Phylum %in% phylums] <- "Other Phyla"

df2 <- select(df, Sample, Phylum, Family, Abundance, group) %>%
  mutate(Phylum=factor(Phylum, levels=c(phylums, "Others")),
         Family=fct_reorder(Family, 10*as.integer(Phylum) + grepl("Others", Family)))
df2$group[df2$group=="Proteobacteria-UnknownFamily"] <- "Proteobacteria-Unknown_Family"
df2$group[df2$group=="Proteobacteria-Unknown_Family"] <-"Proteobacteria-Unknown_Family"
df2$group[df2$group=="Acidobacteriota-Acidobacteriaceae(Subgroup1)"] <-"Acidobacteriota Acidobacteriaceae Subgroup 1"

sort(unique(df2$group))
colours <- c("chartreuse1","chartreuse4", #Acidobacteriota
             "chocolate4", #Actinobacteriota
             "royalblue1", #Bacteroidota
             "seagreen1", #Bdellovibrionota
             "black", #Firmicutes
             "grey", #Other
             plasma(8), #Proteobacteria
             "white") #Unclassified
# Plot
ggplot(df2, aes(x=level_order, y=Abundance, fill=group, order=group)) + 
  geom_bar(aes(fill=group), stat="identity", position="stack") + 
  mytheme + theme(axis.text.x = element_text(angle=75, vjust = 0.5)) +
  scale_fill_manual("", values=colours) +
  scale_x_discrete("") +
  theme(
    axis.text.x = element_text(angle=90, vjust=0.5, hjust=1, size=30, colour = "black"), 
    axis.text.y = element_text(size = 22, face = "bold", colour = "black"),
    axis.title.y = element_text(size = 26, face = "bold", colour = "black")
  ) +  # Vertical x-axis tick labels
  scale_y_continuous("Relative Abundance [%]", breaks=seq(0,100,5), limits = c(0, 101), expand = c(0,2)) +
  guides(fill=guide_legend(nrow=length(unique(df2$group)))) + labs(fill = "Family") +
  geom_vline(xintercept = c(2.5,4.5), color = "black", size=3)

```


